%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
% Masters/Doctoral Thesis 
% LaTeX Template
% Version 2.5 (27/8/17)
%
% This template was downloaded from:
% http://www.LaTeXTemplates.com
%
% Version 2.x major modifications by:
% Vel (vel@latextemplates.com)
%
% This template is based on a template by:
% Steve Gunn (http://users.ecs.soton.ac.uk/srg/softwaretools/document/templates/)
% Sunil Patel (http://www.sunilpatel.co.uk/thesis-template/)
%
% Template license:
% CC BY-NC-SA 3.0 (http://creativecommons.org/licenses/by-nc-sa/3.0/)
%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

%----------------------------------------------------------------------------------------
%	PACKAGES AND OTHER DOCUMENT CONFIGURATIONS
%----------------------------------------------------------------------------------------

\documentclass[
11pt, % The default document font size, options: 10pt, 11pt, 12pt
oneside, % Two side (alternating margins) for binding by default, uncomment to switch to one side
english, % ngerman for German
doublespacing, % Single line spacing, alternatives: onehalfspacing or doublespacing
%draft, % Uncomment to enable draft mode (no pictures, no links, overfull hboxes indicated)
%nolistspacing, % If the document is onehalfspacing or doublespacing, uncomment this to set spacing in lists to single
%liststotoc, % Uncomment to add the list of figures/tables/etc to the table of contents
%toctotoc, % Uncomment to add the main table of contents to the table of contents
%parskip, % Uncomment to add space between paragraphs
%nohyperref, % Uncomment to not load the hyperref package
headsepline, % Uncomment to get a line under the header
%chapterinoneline, % Uncomment to place the chapter title next to the number on one line
%consistentlayout, % Uncomment to change the layout of the declaration, abstract and acknowledgements pages to match the default layout
]{MastersDoctoralThesis} % The class file specifying the document structure

\usepackage[utf8]{inputenc} % Required for inputting international characters
\usepackage[T1]{fontenc} % Output font encoding for international characters

\usepackage{mathpazo} % Use the Palatino font by default

\usepackage[backend=bibtex,style=numeric,natbib=true]{biblatex} % Use the bibtex backend with the authoryear citation style (which resembles APA)

\addbibresource{Legacy.bib} % The filename of the bibliography

\usepackage[autostyle=true]{csquotes} % Required to generate language-dependent quotes in the bibliography

%----------------------------------------------------------------------------------------
%	MARGIN SETTINGS
%----------------------------------------------------------------------------------------

\geometry{
	paper=a4paper, % Change to letterpaper for US letter
	inner=2.5cm, % Inner margin
	outer=2.5cm, % Outer margin
	bindingoffset=.5cm, % Binding offset
	top=1.5cm, % Top margin
	bottom=1.5cm, % Bottom margin
	%showframe, % Uncomment to show how the type block is set on the page
}

%----------------------------------------------------------------------------------------
%	THESIS INFORMATION
%----------------------------------------------------------------------------------------

\thesistitle{Mutational Patterns in Founder Populations} % Your thesis title, this is used in the title and abstract, print it elsewhere with \ttitle
\supervisor{Dr. Simon \textsc{Gravel}} % Your supervisor's name, this is used in the title page, print it elsewhere with \supname
\examiner{} % Your examiner's name, this is not currently used anywhere in the template, print it elsewhere with \examname
\degree{Doctor of Philosophy} % Your degree name, this is used in the title page and abstract, print it elsewhere with \degreename
\author{Luke \textsc{Anderson-Trocm\'e}} % Your name, this is used in the title page and abstract, print it elsewhere with \authorname
\addresses{} % Your address, this is not currently used anywhere in the template, print it elsewhere with \addressname

\subject{Statistical Population Genetics} % Your subject area, this is not currently used anywhere in the template, print it elsewhere with \subjectname
\keywords{} % Keywords for your thesis, this is not currently used anywhere in the template, print it elsewhere with \keywordnames
\university{\href{http://www.mcgill.ca}{McGill University}} % Your university's name and URL, this is used in the title page and abstract, print it elsewhere with \univname
\department{\href{https://www.mcgill.ca/humangenetics/}{Department of Human Genetics}} % Your department's name and URL, this is used in the title page and abstract, print it elsewhere with \deptname
\group{\href{http://simongravel.lab.mcgill.ca/Home.html}{Gravel Lab}} % Your research group's name and URL, this is used in the title page, print it elsewhere with \groupname
\faculty{\href{https://www.mcgill.ca/medicine/}{Faculty of Medecine}} % Your faculty's name and URL, this is used in the title page and abstract, print it elsewhere with \facname

\AtBeginDocument{
\hypersetup{pdftitle=\ttitle} % Set the PDF's title to your title
\hypersetup{pdfauthor=\authorname} % Set the PDF's author to your name
\hypersetup{pdfkeywords=\keywordnames} % Set the PDF's keywords to your keywords
}

\begin{document}

\frontmatter % Use roman page numbering style (i, ii, iii, iv...) for the pre-content pages

\pagestyle{plain} % Default to the plain heading style until the thesis style is called for the body content

%----------------------------------------------------------------------------------------
%	TITLE PAGE
%----------------------------------------------------------------------------------------

\begin{titlepage}
\begin{center}

\vspace*{.06\textheight}
{\scshape\LARGE \univname\par}\vspace{1.5cm} % University name
\textsc{\Large Qualifying Exam}\\[0.5cm] % Thesis type

\HRule \\[0.4cm] % Horizontal line
{\huge \bfseries \ttitle\par}\vspace{0.4cm} % Thesis title
\HRule \\[1.5cm] % Horizontal line
 
\begin{minipage}[t]{0.4\textwidth}
\begin{flushleft} \large
\emph{Author:}\\
{\authorname} % Author name - remove the \href bracket to remove the link
\end{flushleft}
\end{minipage}
\begin{minipage}[t]{0.4\textwidth}
\begin{flushright} \large
\emph{Supervisor:} \\
{\supname} % Supervisor name - remove the \href bracket to remove the link  
\end{flushright}
\end{minipage}\\[3cm]
 
\vfill

%\large \textit{A thesis submitted in fulfillment of the requirements\\ for the degree of \degreename}\\[0.3cm] % University requirement text
%\textit{in the}\\[0.4cm]
\groupname\\\deptname\\[2cm] % Research group name and department name
 
\vfill

{\large \today}\\[4cm] % Date
%\includegraphics{Logo} % University/department logo - uncomment to place it
 
\vfill
\end{center}
\end{titlepage}


%----------------------------------------------------------------------------------------
%	THESIS CONTENT - CHAPTERS
%----------------------------------------------------------------------------------------

\mainmatter % Begin numeric (1,2,3...) page numbering
\pagestyle{thesis} % Return the page headers back to the "thesis" style

\section{Introduction}
Human expansion out of Africa (OOA) is estimated to have occurred between 45,000 and 60,000 years ago (Henn, 2012). 
As humans expanded into new habitats, each new colony was founded by a smaller group departing a larger, more diverse source population.
The impacts of the OOA bottleneck are still measurable in modern human populations, with distance from Africa being strongly correlated with decreasing genetic diversity (Ramachandran, 2005).


Bottlenecks can drastically change the allele frequencies within a population and lead to rapid divergences (Rosenberg, 2002; Charlesworth, 2003).
Rare alleles in the source population can increase significantly in frequency or even become fixed in the founder population.
Isolation over time can lead to increased diversity between small populations, while the diversity within may remain relatively low. 
For this reason, migration between populations can increase the genetic diversity of both populations.


Recent events of admixture are tractable with genomic analyses like identity by descent (IBD) mapping and coalescence estimations (Baharian, 2016).
However, recombination over time will fragment haplotypes eventually making it difficult to determine their ancestry.
Despite this, a population will retain a unique set of alleles at frequencies waxing and waning over time.
These allele frequencies can be used to estimate demographic histories of multiple populations (Gravel, 2010). 

Many disease or phenotype association tests are susceptible to spurious results caused by population structure.
In order to properly account for this, a better understanding of human demographic history is needed (Gravel, 2010).
Moreover, studying recent founder populations can lead to better understanding of the mechanisms and dynamics of historical founder events. 

%The last 5 years have seen a drastic increase in the amount and quality of human genome sequence data.
%As a result of decreasing costs per sample, research centres are now able to perform genomic studies involving larger and larger cohorts.
%Need, A. C. & Goldstein, D. B. Next generation disparities in human genomics: concerns and remedies. Trends Genet. 25, 489?494 (2009).
%These large scale genomics studies have increased statistical power to a point where we can resolve fine scale mutational signatures observable on evolutionary scales by comparing the frequencies of mutations between populations.
%These mutational patterns are useful for inferring population demographic histories, as well as variation in the human mutation rate.


%Estimates of the parameters of this OOA bottleneck using sequence data and modelling population dynamics range from 2,500 to as low as 450 individuals founding Europe and Asia (Eller, 2004; Fagundes, 2007).
\subsection{Thesis proposal}

The aim of this PhD project is to study the genomics of founder populations to infer mutational signatures, demographic history, population structure and mutation rates.
\textbf{Our hypotheses are }: \textbf{1}. An unusual mutational patterns observed in Japan is not reproducible and is the result of a technical artifact.
\textbf{2}. Quebec has more structure than would be expected from a series of founder events which can be accounted for by ancestral population structure predating the founder events.
\textbf{3}. Leveraging the deep phylogenetic records from Quebec in combination with sequence data can improve the estimates of human mutation rates.

%Combining genetic, genealogical and geographic data can explain the population structure observed today, recreating historical demographics.
%This project will be broken down into three chapters :
%1. Unique signature in Japan
%Identify the source of the mutational pattern in Japanese population. 
%2. Characterize the genetic diversity of the Quebec population (a case study)
%Explain population structure. Using Fst and UMAP/GEO.
%3. Explore mutation rate in Quebec population
%apply existing methods of estimating human mutation rate.
%can we do better? combine data to improve estimation. 

\chapter{A curious mutational signature in Japan}

The last 5 years have seen a drastic increase in the amount and quality of human genome sequence data. 
Reference cohorts such as the International HapMap Project \citep{HapMap2005}, the 1000 Genomes Project \citep{1000GenomesProjectConsortium2010,The1000GenomesProjectConsortium2012}, and the Simons Diversity project \citep{Mallick2016}, for example, have made thousands of genome sequences publicly available for population and medical genetic analyses. 
Many more genomes are available indirectly through servers providing imputation services \citep{HapMap2005} or summary statistics for variant frequency estimation \citep{Lek2016}.

The first genomes in the 1kGP were sequenced 10 years ago \citep{VanDijk2014}. 
Since then, sequencing platforms have rapidly improved. 
Yet, because of the extraordinary value of freely available data, early data from the 1000 Genomes project (1kGP) is still widely used as a reference panel for imputation, allele frequency estimations and to answer a wide range of medical and evolutionary questions. 
The second phase of the 1kGP implemented multiple technological and analytical improvements over its earlier phases \citep{The1000GenomesProjectConsortium2012}, leading to heterogeneous sample preparations and data quality.

Even though such batch effects clearly have the potential to confound analyses, the usefulness and ubiquity of the 1kGP data make such issues particularly insidious.   
This raises the question of whether and how such legacy data should be included in contemporary analyses alongside more recent cohorts.
Here we point out how large and previously unreported batch effects in the early phases of the 1kGP still lead to incorrect genetic conclusions through population genetic analyses and indirect use through prominent imputation servers.


\subsection{Mutational Signatures}
There has been recent interest in population variation in germline mutational signatures which can be revealed in large sequencing panels. 
In 2015, Harris reported 50\% more TCC ${\rightarrow}$ TTC mutations in European populations compared to African populations, and this was replicated in a different cohort in 2017 \citep{Harris2015a, Harris2017a, Mathieson2017a}. 
Strong population enrichments of a mutational signature suggests important genetic or environmental differences in the history of each population \citep{Harris2015a, Harris2017a}. 
Harris and Pritchard further identified distinct mutational spectra across a range of populations, which were further examined in a recent publication by Aikens et al. \citep{Harris2017a,Aikens2018}.
 
In particular, both studies  identified a heterogeneous mutational signature within 1000 Genomes Japanese individuals.
This heterogeneity is intriguing because differences in germline signatures accumulate over many generations.
A systematic difference within the Japanese population would suggest sustained environmental or genetic differences across sub-populations within Japan with little to no gene flow.
We therefore decided to follow up on this observation, by using a newly sequenced dataset of Japanese individuals. 

\section{Results}
Harris and Pritchard reported an excess of a 3-mer substitution patterns *AC${\rightarrow}$*CC in a portion of the Japanese individuals in the 1kGP \citep{Harris2017a}.
While trying to follow up on this observation in a larger and more recent Japanese cohort, we did not find this particular signature.
When comparing the allele frequencies between the Japanese individuals from the 1kGP and this larger dataset, we observed a number of single nucleotide polymorphisms (SNPs) private to one of the two groups.
Given the similarity of the two populations, this strongly suggests a technical difference rather than a population structure effect.
These mismatches were maintained after filtering for low-quality regions of the human genome and sites failing Hardy-Weinberg equilibrium tests.

To identify SNPs that are likely to reproduce poorly across cohorts without having access to a second cohort, we performed an association study in the JPT for SNPs that associate strongly with low $Q$ (Figure \ref{SFS}).
Traditionally, genome wide association studies use genotypes as the independent variable. 
Here we perform a "reverse GWAS", in the sense that genotypes are now the dependent variable that we attempt to predict using the continuous variable $Q$ as the independent variable.
We use logistic regression of the genotypes on $Q$ and identify 587 SNPs with $p < 10^{-8}$ and 1034 SNPs with $ p < 10^{-6}$.
The variants that are associated to $Q$ have an enrichment in *AC${\rightarrow}$*CC mutations, GA*${\rightarrow}$GG*, and GC*${\rightarrow}$GG* mutations.
These three enrichments can be summarized as an excess of G**${\rightarrow}$GG* in individuals with low $Q$.

The distribution of $Q$ across 1kGP populations shows that many populations have distributions of $Q$ scores comparable to that of the JPT, especially populations sequenced in the phase 1 of the project: sequencing done in the early phases of the 1kGP was more variable and overall tended to include lower quality sequencing data (Figure \ref{MapQual}).
This variability could result from evolving sequence platform and protocols or variation between sequencing centres. 
By 2011, older sequencing technologies were phased out, and methods became more consistent, resulting in higher and more uniform quality.
We performed the reverse GWAS approach in all populations and identified 3826 variants that were independently associated to low $Q$ in at least two populations with $ p < 10^{-6}$ (Figure \ref{OverLap}). 

\chapter{Characterization of the Quebec population}

\section{Brief History of Quebec}
The Quebec population was founded over four centuries ago from around 8500 settlers from France.
This small number of settlers are the direct ancestors of the great majority of Quebec's population of 7 million.
In population genetics, a founder effect is characterized by a small number of founders from a larger source population giving rise to a population expansion.[***NEEDS WORK***]
This tends to increased homozygosity and rapidly shift allele frequencies compared to the source population.

The north east of Quebec (e.g. Saguenay-Lac-Saint-Jean and Charlevoix) displays higher rates of some rare genetic diseases and increased homozygosity which is suggestive of a severe founder effect. (Gagnon, 2000)
Western Quebec, on the other hand, has similar prevalence of diseases compared to Europe. (Gagnon, 2000)
This discrepancy of disease rates across the Saint-Lawrence valley indicates that the severity of the bottleneck varies across geographic regions. (Gagnon, 2000)

Indeed, the intensity of the founder effect in Quebec is not consistent across the population as there is evidence of geographical stratification which has been attributed to a number of factors like varying concentration of founders, native ancestry and geographic isolation. (Gagnon, 2000).
The founders of Quebec were not all from the same regions in France, providing some initial diversity despite a small initial population size.
Moreover, Native American admixture with founders has been estimated to be around 1\% which would have also been a source of genetic diversity among these settlements. (Moreau, 2013)
Some settlements were more isolated than others resulting in less gene flow between sub-populations.
The north-east of Quebec is characterized historically as being more isolated with higher emigration rates compared to western Quebec.

\subsection{Quebec : A case study of founder populations}

The Quebec founder effect is an interesting case study for population genetics for multiple reasons.
Firstly, the source population of this bottleneck is still present, which may not be the case for older founder populations. 
This provides us with the possibility of comparing allele frequencies between Quebec and its ancestral population.
Secondly, many populations experienced founder effects as humans expanded out of Africa and populated the continents.
Studying examples of founder effects in human populations are a great way to improve our understanding of human demographic histories.
Finally, one unique aspect of this population is the deep genealogical data that spans four centuries and includes nearly 3 million entries.
Quebec is not the only recent founder population, but it is among the largest and in combination with its rich genealogical records, it makes it an excellent cohort to study the impact of founder events.

There have been several studies in the past few decades that have examined the genetic diversity and population structure of Quebec.
Using genotype data, linkage disequilibrium and homozygosity analyses of sub-populations within Quebec showed significant structure and relatedness that persists in the current population (Roy-Gagnon, 2011).
Quebec is not a homogeneous population, but rather made up of a combination of distinct sub-populations.
This was confirmed with Fst statistics of these sub-populations, indicating substantial genetic structure.
Of particular interest, they reported that there was greater differentiation between sub-populations of Quebec than between European populations.
While this study included 140 genotyped individuals - a small sample by today's standards - they postulate that this difference is accounted for by genetic drift.
Indeed, the heterogeneity of the founder effect in the Quebec population could be explained by a series of separate founder events influenced by variation in the relative isolation, and migration between settlements.

\section{Genomic and genealogical data}

The second chapter of my thesis will examine the genetic structure of the Quebec population using more recent and larger genomic datasets in combination with the deep-genealogical and geographical data.

While the deep-genealogical data has been studied for over a decade, linking the genealogy to genetic data has only been done with relatively small sample sizes. 

Genizon data has 000 samples, genotyped using a variety of chips. These data require some cleaning and processing before analysis. This cohort has yet to be published. 

We will also include data from the Cartagene project which has 000 samples from...

With these data, we hope to reproduce the results from previous work and answer the following question :
(2) Are the sub-populations of Quebec the result of genetic drift from founder events and isolation, or the preservation of genetic structure from France?

\subsection{approach and methods}
Using Fst statistics we can infer the relative differences between sub-populations of Quebec, as done in (Roy-Gagnon, 2011).
Dominic Nelson has written software that can produce forward simulations of the Quebec population using the genealogical data from BALSAC.
These simulations can be used to produce estimated differences between sub-populations resulting purely from a founder effect.

Comparing the Fst statistics between the real data and simulations will determine if the structure observed in Quebec is the result of preserved genetic structure from France or the result of drift from founder events.

\section{Linking population structure to geography (results)}
Preliminary analyses of the Genizon dataset are in agreement with previous work.
In collaboration with Alex Diaz-Papkovich, we performed a UMAP projection of the Genizon genotype data.
This method takes the first *10* principle components of the data as input and reduces this high dimensional data preserving both local and global structure.
This projection shows significant population structure within the Quebec population. 
These data cluster into 5 roughly defined sub-populations, in agreement with what (Roy-Gagnon, 2011) observed \ref{Cluster}. 


We colored these plots based on the three dimensional (X,Y,Z) coordinates of the UMAP projection space converted to three dimensional color space (R,G,B).
Thus, points that are close together in UMAP space will have similar color.

We were able to link 000 of the Genizon individuals to the BALSAC database which includes marriage location of their ancestors.
Combining the ancestral locations of these individuals with the three dimensional UMAP coloring scheme, geographic clustering of the sub-populations becomes apparent \ref{UMAP}.  

These preliminary results are very promising as they appear to capture the geographic and genetic clustering of the Quebec population.





\section{Future Directions}
%We know that there are regional differences in disease rates that are linked to population substructure.
%The genetic data used in UMAP analysis also identify similar clusters.
%Identifying the variants that are responsible for the clustering could also provide us with a shortlist of variants that could also be linked to the diseases that are associated with these sub-populations.
%This method has yet to be formalized, but in brief, performing a logistic regression predicting presence of genotypes using the axis of variation that separates one group from the others could be used to identify these variants.
While Fst gives a good estimate of genome wide past coalescent rates, the analysis of long IBD segments can provide us with more detailed insight of more recent demographic history (Ralph, 2013; Charlesworth 2003).
Because Fst takes in all polymorphisms as input for the statistic, it does not distinguish between recent and more ancient common ancestry.
Analysis of IBD segments shared among individuals from Quebec can be useful for reconstructing the recent structure of relatedness.
The length of the shared segments can be used to infer the number of generations leading back to a common ancestor.
Combining this with geographic data can infer patterns of shared ancestry across the Saint-Lawrence valley.
Ralph (2013) identified a general trend of decreased of shared ancestry with geographic distance as well as more regional signals of IBD in European populations.
Europe has a rich history of complex migratory patterns and demographic histories that is difficult to trace back with certainty. 
Quebec on the other hand has a much more recent history that may be easier to trace with IBD analysis.

While studying Quebec populations on their own can provide detailed historical inference of demographics, including samples from other European populations can provide more context.
Expanding the current analyses to include data from French individuals could be used to confirm the results from the Fst analysis as well as provide insight into which regions in France are source populations to regions in Quebec.
French genome data is obtainable through Dr. Christian Dina, one of our collaborators in the University of Nantes. 

\chapter{Estimating mutation rates}

Mutation rate estimations are necessary for understanding the timing of evolutionary processes and events.
Neutral mutations are expected to arise at a constant rate in the genome, and can therefore be used as an evolutionary clock that can be used to estimate divergence times (Moorjani, 2016). 
The basic equation of mutation rate is $N / LM$ where N is the number of mutations, L is the length of the region and M is the number of generations. (Narasimah, 2016)
Estimating mutation rates in humans has proven to be difficult for a number of reasons.
Mutation rates vary across the genome as some regions are more conserved than others. 
Mutation rates also vary over time.
Shorter branches within the Ape family compared to other primates suggests a "hominoid rate slowdown" whereby mutation rates decreased in apes. (Moorjani, 2016; Scally, 2012)
More recent studies have found that mutation rates vary among human populations. (Harris, 2015, Harris Pritchard2017)
Converting the mutation rate per generation to a yearly rate is also met with difficulties when estimating average human generation time.
Generation times are known to vary among current populations, and likely varied over longer time spans (Henn, Moorjani, 2016)

\section{Review of current methods and limitations}

The first human mutation rates were estimated by Haldane (1935) using the rates of children born with rare diseases from unaffected parents.
However, once sequence data became available, estimates were based on phylogenetic data and fossil records.
By using fossil record dating as an estimate for coalescence time $t$ and the sequence divergence $d$ between two species as the number of mutation that have occurred over that time span, mutation rates $\mu$ can be estimated as $\mu = 2d/t$ (Scally, 2012). 
Using this method with the time of divergence of Chimps and Humans, the mutation rates are estimated to be around $1\times10^{-9}$ up to $2.5\times10^{-8}$ (Takahata, 1997; Scally, 2012; Moorjani, 2016, Narasimah, 2016).
The limitations of this approach are numerous, ranging from accuracy of fossil record dating, to estimating the average generation time for chimps and humans over millions of years.

Instead of using distant ancestry and fossil records, another more direct approach to estimating mutation rates is to sequence the genomes of parents and offspring. 
This approach tends to give lower estimates compared to phylogenetic approach ($1-1.25\times10^{-8}$) (Narasimah, 2016).
This rate is less than half of the rates estimated from phylogenies (Shendure, 2015). 
One issue with trio sequencing for mutation rates is that false-positive rates of sequencing technologies are difficult to account for considering how low the true mutation rate is.
Another issue is that the samples used for sequencing are not zygotes.
This leaves room for germline mosaicisms to artificially increase the number of mutations observed in a generation.

Most recently, methods for estimating mutation rates using population genetics are increasingly being used.
These approaches expand on the theory behind trio sequencing by including more distantly related individuals.
One approach involves autozygous segments observed in a founder population with a known pedigree (Campbell, 2012).
These segments shared by multiple individuals are compared and mutations in these segments are then used in combination with the number of generations separating two individuals (Campbell, 2012).
Palamara et al. (2015) describe an even more generalized method by identifying IBD segments and inferring number of generations using coalescent theory (Baharian 2016).
This method then regresses the number of mutations observed per base against the time of most recent common ancestor.
The slope of the regression is the mutation rate per generation.
Applying this method to 498 individuals from 250 families, they measured a mutation rate of ($2.08\times10^{-8}$).
This method is able to infer the genotyping false-positive rate as the intercept of the regression line.
The major limitation to this method comes from the coalescence estimations.

\subsection{Apply state of the art methods to new data}
For the third chapter of my thesis I plan on exploring mutation rate estimations using the combination of sequence data and phylogenetic data from the Quebec population.
One preliminary objective might be to replicate the population genetics methods described above by Palamara et al. and Campbell et al..
Knowing that mutation rates can vary across populations, applying these methods to a new dataset may in itself be worthy of publication.

Considering the deep phylogenetic data available for the Quebec population, we may be able to improve on the Palamara method by acquiring more accurate estimates of generation time separating two IBD segments.
Indeed, by linking individuals to the phylogenetic tree of the Quebec population, we could estimate the most recent common ancestor for a given segment of the genome.
By grouping IBD segments according to genomic regions (e.g. coding, CpG, etc.) and running the analysis separately would provide us with a more detailed view of the genome wide mutation rate in humans.




%----------------------------------------------------------------------------------------
%	THESIS CONTENT - APPENDICES
%----------------------------------------------------------------------------------------

\appendix % Cue to tell LaTeX that the following "chapters" are Appendices

\begin{figure}
\includegraphics[width=\hsize,keepaspectratio]{./Figures/Genizon_BalSac_mapsInMaps4.jpg}
\label{UMAP}
\end{figure}

\begin{figure}
\includegraphics[width=\hsize,keepaspectratio]{./Figures/5Clusters.png}
\label{Cluster}
\end{figure}

% Include the appendices of the thesis as separate files from the Appendices folder
% Uncomment the lines as you write the Appendices

%\include{Appendices/AppendixA}


%----------------------------------------------------------------------------------------
%	BIBLIOGRAPHY
%----------------------------------------------------------------------------------------

\printbibliography[heading=bibintoc]

%----------------------------------------------------------------------------------------

\end{document}  
