---
title: "Comparing Results with Mafessoni et al."
author: 
- name: Luke Anderson-Trocmé
  affiliation:
  - &cruk Department of Human Genetics, McGill University,  Montreal, Canada
date: '`r format(Sys.Date(), "%Y-%B-%d")`'
output:
  html_notebook:
    df_print: paged
    code_folding: show
    toc: yes
    toc_float: 
      collapsed: false
      smooth_scroll: false
---

load libraries
```{r, message = FALSE}
library(ggplot2)
library(data.table)
library(tidyverse)

#Directories
DataPath <- '~/Documents/LegacyData/PlottingScripts/ProcessedData/'
figurePath <- '~/Documents/LegacyData/Figures/'
```

define functions
```{r, include = FALSE}
ReverseComp <- function(x)
        chartr("ATGC","TACG",
        sapply(
          lapply(
          strsplit(x, NULL), 
          rev), 
          paste, 
          collapse = ""))

processContext <- function(data){
  
#get reverse complement of context, ref and alt
data$Rev<-ReverseComp(as.character(data$Context))
data$RevRef<-ReverseComp(as.character(data$Ref))
data$RevAlt<-ReverseComp(as.character(data$Alt))
#only get one half (to fold over)
AC<-data[which(data$Ref 
               %in% 
                 c('A','C')),]
TG<-data[which(data$Ref 
               %in% 
                 c('T','G')),]
#fold over
TG$Context <- TG$Rev
TG$Ref <- TG$RevRef
TG$Alt <- TG$RevAlt
#bind them together
data <- rbind(AC,TG)
#combine it all together to get a context
data$Mut <- paste0(data$Ref,'->', 
                substr(data$Context,1,1),
                data$Alt,
                substr(data$Context,3,3))
#get the counts of each bin
binCount <- as.data.frame(table(data$Mut))
#process for plotting
binCount$Start <- substr(binCount$Var1,4,4)
binCount$End <- substr(binCount$Var1,6,6)
binCount$Mut <- paste(substr(binCount$Var1,1,3),
                      substr(binCount$Var1,5,5))
binCount$Mut <- gsub('->', ' →',binCount$Mut)

binCount$Var1 <- NULL

return(binCount)
}


plotContext <- function(data, name){
  
plt <- ggplot(data, 
              aes(x = End, 
                  y = Start, 
                  fill = Freq)) +
    geom_tile() +
    facet_grid(.~Mut) +
    labs(y = name) +
    theme_classic() +
    theme(axis.title.x = element_blank(),
          axis.line = element_blank(),
          axis.text = element_text(size = 20),
          axis.title.y = element_text(size = 20),
          strip.placement = "outside",
          strip.background = element_blank(),
          strip.text = element_text(size = 20),
          legend.text = element_text(size = 20),
          legend.title = element_text(size = 20),
          panel.border = element_rect(colour = "black", 
                                      fill = NA, 
                                      size = 1),
          plot.margin = unit(c(0,0,0,1.5), "lines")) +
    scale_fill_gradient(limits = c(0, maxFreq),
                        low = 'white', 
                        high = 'black',
                        breaks = c(100,250,500),
                        guide = guide_legend(title = 'Variant\ncount',
                                             direction = 'vertical'))
return(plt)
}
```

load data
```{r}
#load regression results
Regression <- readRDS('/Users/luke/Documents/Legacy_Misc/Misc/Regression.Rmd')
#load supplementary files from Mafessoni
#coding
t4 <- read.table('/Users/luke/Downloads/evy199_supp/TableS4.bed', comment.char = "@", header = T)
t4$type <- "rare_coding"
t5 <- read.table('/Users/luke/Downloads/evy199_supp/TableS5.bed', comment.char = "@", header = T)
t5$type <- "common_coding"
#intergenic
t6 <- read.table('/Users/luke/Downloads/evy199_supp/TableS6.bed', comment.char = "@", header = T)
t6$type <- "common_intergenic"
t7 <- read.table('/Users/luke/Downloads/evy199_supp/TableS7.bed', comment.char = "@", header = T)
t7$type <- "rare_intergenic"

#get total unique positions from most conservative tables
totalT<- 
  rbind(t4,t5,t6,t7) %>%
  group_by(X.chr, end) %>% 
  top_n(1) #keep unique positions only

merged_regression <- merge(Regression, totalT ,by.x = c('Chr','Pos'),by.y=c('X.chr','end'), all.x=T)
merged_regression[is.na(merged_regression)] <- 1

#remove positions that don't pass strict mask
merged_regression_masked <- 
  anti_join(merged_regression, 
            strictMask,
            by = c('Chr','Pos','rsID'))

#remove positions that DO pass strict mask
merged_regression_unmasked <- 
  anti_join(merged_regression, 
            merged_regression_masked)

nrow(merged_regression)

nrow(merged_regression_unmasked)

nrow(merged_regression_masked)

merged_regression_unmasked$strict_mask <- "Pass strict mask"
merged_regression_masked$strict_mask <- "Fail strict mask"

merged_regression <- rbind(merged_regression_masked, merged_regression_unmasked)
```

```{r}
#select rows of interest
#either present in Mafessoni OR significant in our data

to_plot <- merged_regression %>%
           dplyr::filter(pvalue_glm_filt < 1 | Adjusted_Log10p > 2)

p_comp <- 
  ggplot(to_plot, 
         aes(x=Adjusted_Log10p, 
             y= -log10(pvalue_glm_filt), 
             color = strict_mask)) +
  geom_point(shape = 1,
             size = 1,
             alpha = 0.3) +
  labs(x = '-log10(p) Q-association',
       y = '-log10(p) LD-association') +
  scale_color_grey(start = 0.1, end = 0.6) +
  guides(color = F) +
  facet_wrap(.~strict_mask)+
  theme_classic() +
  theme(axis.title = element_text(size = 20),
        axis.text = element_text(size = 20))

p_comp
```
```{r}
#missing in Mafessoni
p <- to_plot %>%
  dplyr::filter(strict_mask == paste0("Pass strict mask (",nrow(merged_regression_unmasked),")"))

m_maf <- p %>%
           dplyr::filter(pvalue_glm_filt == 1 & Adjusted_Log10p > 2)

#missing in ours
m_us <- p %>%
           dplyr::filter(pvalue_glm_filt < 1 & Adjusted_Log10p < 2)

#present in both
both <- p %>%
           dplyr::filter(pvalue_glm_filt < 1 & Adjusted_Log10p > 2)

print(paste("missing in Mafessoni (Pass): ",nrow(m_maf), "or : ", nrow(m_maf)/nrow(p)*100) )
print(paste("missing in ours (Pass): ",nrow(m_us), "or : ",nrow(m_us)/nrow(p)*100))
print(paste("present in both (Pass): ",nrow(both), "or : ",nrow(both)/nrow(p)*100))
print(paste("total (Pass): ", nrow(p)))

#missing in Mafessoni
f <- to_plot %>%
  dplyr::filter(strict_mask == paste0("Fail strict mask (",nrow(merged_regression_masked),")"))

m_maf <- f %>%
           dplyr::filter(pvalue_glm_filt == 1 & Adjusted_Log10p > 2)

#missing in ours
m_us <- f %>%
           dplyr::filter(pvalue_glm_filt < 1 & Adjusted_Log10p < 2)

#present in both
both <- f %>%
           dplyr::filter(pvalue_glm_filt < 1 & Adjusted_Log10p > 2)

print(paste("missing in Mafessoni (Fail): ",nrow(m_maf), "or : ", nrow(m_maf)/nrow(f)*100) )
print(paste("missing in ours (Fail): ",nrow(m_us), "or : ",nrow(m_us)/nrow(f)*100))
print(paste("present in both (Fail): ",nrow(both), "or : ",nrow(both)/nrow(f)*100))
print(paste("total (Fail): ", nrow(f)))


m_maf <- to_plot %>%
           dplyr::filter(pvalue_glm_filt == 1 & Adjusted_Log10p > 2)

#missing in ours
m_us <- to_plot %>%
           dplyr::filter(pvalue_glm_filt < 1 & Adjusted_Log10p < 2)

#present in both
both <- to_plot %>%
           dplyr::filter(pvalue_glm_filt < 1 & Adjusted_Log10p > 2)

print(paste("missing in Mafessoni (total): ",nrow(m_maf), "or : ", nrow(m_maf)/nrow(to_plot)*100) )
print(paste("missing in ours (total): ",nrow(m_us), "or : ",nrow(m_us)/nrow(to_plot)*100))
print(paste("present in both (total): ",nrow(both), "or : ",nrow(both)/nrow(to_plot)*100))
print(paste("total (total): ", nrow(to_plot)))
```

```{r}
mafessoni_snps <- 
  totalT[,c('X.chr','end')]

write.table(mafessoni_snps, 
            file = paste0(DataPath,
                        'mafessoni_snps.pos.txt'), 
            col.names = FALSE, 
            row.names = FALSE, 
            quote = FALSE, 
            sep = '\t')

```

extract context for Mafessoni SNPs
```{bash}
#Using GenomeWide.AncestralContext.txt
#To get mutation context, use $path/AncestralContext.py
#compare first two columns in both files, extract matches
out_path='/Users/luke/Documents/LegacyData/LatexManuscript/SupplementaryData/'
path='/Users/luke/Documents/LegacyData/PlottingScripts/ProcessedData/'

if [ ! -f $path/mafessoni_snps.pos.context.txt ]; then
  echo "pos context"
  awk 'NR==FNR{a[$1,$2];next} ($1,$2) in a' \
  $path/mafessoni_snps.pos.txt \
  $path/GenomeWide.AncestralContext.txt \
  > $path/mafessoni_snps.pos.context.txt
fi  
```

```{r}
contextHeader = c('Chr','Pos','Ref','Alt','Flip','Context')

#load SNP context
mSNP <- read.table(paste0(DataPath, 
                          'mafessoni_snps.pos.context.txt'), 
                sep = '\t', 
                header  = FALSE,
                col.names = contextHeader)

mSNP <- processContext(mSNP)

maxFreq <- max(mSNP$Freq)
mSNPplt <- plotContext(mSNP,'LD-associated')


m_plot <-
  plot_grid(p_comp,
          mSNPplt,
          ncol = 1,
          rel_heights = c(4, 1))

ggsave(paste0(figurePath,
              'Mafessoni_MutSpect.jpg'),
       m_plot, 
       height = 10,
       width = 10)

m_plot
```




```{r}


```