---
title: "HGP analysis"
author: 
- name: Rick Farouni
  affiliation:
  - &cruk Department of Human Genetics, McGill University,  Montreal, Canada
date: '`r format(Sys.Date(), "%Y-%B-%d")`'
output:
  html_notebook:
    df_print: paged
    code_folding: show
    toc: yes
    toc_float: 
      collapsed: false
      smooth_scroll: false
---

# Preparations

```{r setup}
knitr::opts_knit$set(root.dir = rprojroot::find_rstudio_root_file(), fig.width=16)
options(scipen=999)
```

```{r}
library(Matrix)
library(irlba)
library(threejs)
library(tidyverse)
library(lfa)
library(furrr)
library(broom)
options(future.globals.maxSize= 10000000000)
plan(multicore)
```

# Download data


Download a fast parser program (after compilation you'll have a program called a.out)

```{bash}
wget https://raw.githubusercontent.com/bwlewis/1000_genomes_examples/master/parse.c
cc -O2 parse.c
```

Download 1000 genomes phenotype data file
 
```{bash}
wget ftp://ftp.1000genomes.ebi.ac.uk/vol1/ftp/technical/working/20130606_sample_info/20130606_g1k.ped
```

Download 1000 genomes genotype data file for chr 20


```{bash}
wget ftp://ftp-trace.ncbi.nih.gov/1000genomes/ftp/release/20130502/ALL.chr20.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.vcf.gz

```

Read just the header of the chromosome file to obtain the sample identifiers


```{r}
ids <- readLines(pipe("zcat ALL.chr20.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.vcf.gz  |
                      sed -n /^#CHROM/p | 
                      tr '\t' '\n' | 
                      tail -n +10"))
head(ids)
```


Write ids to file

```{r}
writeLines(ids, "ids.txt")
```


Download bas files for all the samples 

```{bash message=FALSE}
for id in `cat ids.txt`
  do
  wget ftp://ftp.1000genomes.ebi.ac.uk/vol1/ftp/phase3/data/${id}/alignment/${id}.chrom20.*.bam.bas
done
wait
```

# Read bas files

```{r}
read_basfiles <- function(bas_filepath) {
  
  bas_file <- read_tsv(bas_filepath,
                  col_types = cols_only('sample' = 'c', 
                                        'platform' = 'c',        
                                       'average_quality_of_mapped_bases' = col_guess())) 
  return( bas_file)
}
```

```{r}
bas_files <- list.files(path = ".",
                        pattern="*bam.bas")

bas_dt <- 
  map_dfr(bas_files,read_basfiles ) %>%
  rename(avg_qual="average_quality_of_mapped_bases")
bas_dt
```


```{r}
bas_dt <-
  bas_dt %>%
  group_by(sample) %>%
  mutate(id=1:n()) %>%
  select(-platform)%>%
  spread(id,avg_qual)
bas_dt
```
```{r}
n_not_na <- function(x) {
 sum(!is.na(x))
}
```


```{r}
 bas_dt <- 
  bas_dt %>%
  ungroup()%>% 
    mutate( avg_qual_sd =pmap_dbl(select(., -sample), lift_vd(sd,na.rm =TRUE )),
           avg_qual_mean = rowMeans(select(., -sample),na.rm =TRUE),
           avg_qual_n= pmap_dbl(select(., -sample), lift_vd(n_not_na)),
           avg_qual_min= pmap_dbl(select(., -sample), lift_vd(min, na.rm =TRUE)),
           avg_qual_max= pmap_dbl(select(., -sample), lift_vd(max, na.rm =TRUE))) %>%
  select (sample, avg_qual_n, avg_qual_min,avg_qual_max,avg_qual_mean,avg_qual_sd, everything() ) 
```



```{r}
ped <- 
  read_tsv(url("ftp://ftp.1000genomes.ebi.ac.uk/vol1/ftp/technical/working/20130606_sample_info/20130606_g1k.ped"))%>% 
  rename(sample="Individual ID")
ped 
```



```{r}
bas_dt <- inner_join( ped  %>% 
                       select(sample, Population),
                       bas_dt,
                     by="sample")
bas_dt
```


```{r}
write_tsv(bas_dt, "average_quality_chrom21.txt")
```

 Plot 

```{r fig.width=20}
ggplot(bas_dt,
       aes(x=avg_qual_min,
           y=avg_qual_max, 
           colour=avg_qual_n )) +
  geom_point(size=2)
 
```



```{r fig.width=20}
ggplot(bas_dt,
       aes(x=avg_qual_mean,
           y=avg_qual_sd, 
           colour=avg_qual_n )) +
  geom_point(size=2)
 
```


# Logistic factor analysis

```{r}
bas_dt <- read_tsv( "average_quality_chrom21.txt")
avg_qual_mean <- bas_dt$avg_qual_mean
```

```{r}

to_datalist <- function(i, data_genotype, data_iaf){
  
    data <- tibble(genotype=data_genotype[i,],
               iaf=data_iaf[i,])
               
     return(data)       
}

get_data <- function(){
  
  p <- pipe("zcat ALL.chr20.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.vcf.gz  |
          sed /^#/d  |
          cut  -f '10-' |
          ./a.out | 
          cut -f '1-2'")
  x <- read.table(p,
                colClasses=c("integer","integer"), 
                fill=TRUE,
                row.names=NULL)

  
  data <- sparseMatrix(i=x[,2], j=x[,1], x=1L)
  
  
  return(data)
  
}
  
  
get_af_data <- function(data, d=4){
  

  # compute the first d factors including the intercept
  LF <- lfa(data, d)
  # compute the individual-specific allele frequencies
  data_iaf <- af(data, LF)
  loci <- c(1:dim(data)[1])
  names(loci) <- loci
  
  data_list <- future_map(loci, 
                 to_datalist,
                 data_genotype=data,
                 data_iaf=data_iaf) 
  
  data_list <- 
  enframe(data_list,
          name = "locus", 
          value = "dataset")
  
  return(data_list)
  
}

get_predicted_mean <- function(model_fit) {
 predicted <- 
   augment(model_fit, type.predict = "response") %>%
   select("genotype","offset.iaf.",".fitted") %>%
   rename(iaf="offset.iaf.", mu=".fitted" )
  return(predicted)
}

get_stats <- function(model_fit){
  stats_dt <- 
    tidy(model_fit)  %>%
    filter(term=="avg_qual_mean") %>%
    select(estimate, p.value) %>%
    mutate(deviance= anova(model_fit)["Deviance"][[1]][2])
  return(stats_dt)
}



fit_model <- function(dataset){
  
   fit <- glm(genotype ~ avg_qual_mean + offset(iaf),
                           family ="binomial",
                           data=dataset)
   
   stats <- get_stats(fit)
   predicted <- get_predicted_mean(fit)
   
   return(list(stats, predicted))
}

get_plot <- function(data){
  
 p <- ggplot(bind_cols(data, 
                   bas_dt %>% 
                     select(Population,
                            avg_qual_mean))) +
    geom_point(aes(avg_qual_mean, 
                   genotype),
               size=.5,
               alpha= 0.4) +
    geom_line(aes(avg_qual_mean,
                  iaf, 
                  colour="af"),
              linetype="dashed") +
    geom_line(aes(avg_qual_mean,
                  mu,
                  colour="mu")) +
      theme_bw() +
    facet_wrap(~Population)
  
  return(p)
}
```

```{r}
run_workflow <- function(loci){

  data <- get_data()
  data <- t(as.matrix(data[,loci]))
  data_list <- get_af_data(data) 
  
  rm(data)
  gc()
  
  data_list <- 
    data_list %>%
  slice(1:10) %>%
  transmute(
    locus=locus,
    fit1 = future_map(dataset, fit_model))  %>%
  transmute(
    locus=locus,
    stats = future_map(fit1, c(1)),
    predicted =future_map(fit1, c(2))) %>%
    unnest(stats) %>%
    arrange(-deviance) 
  
  return(data_list)
  
}


```


```{r}
data_list <- run_workflow (loci=c(1:50000))
data_list
```

```{r}
n <-5
target_loci <- 
  data_list %>% 
  slice(1:n) %>% 
  pull(locus)
target_loci 
```

```{r}
data_list_subset <- 
  data_list %>%
   filter(locus %in% target_loci ) %>%
    mutate(plotted= future_map(predicted, get_plot)) 
```



```{r}
data_list_subset %>%
  pull(plotted)
```











```{r}
sessionInfo()
```










## Draft



The matrix has 2,504 rows (people) and 1,812,841 columns (variants)




Determine genome-wide goodness-of-fit for a particular value of d by computing SNP-by-SNP goodness-of-fit when compared to population structure. 

```{r}
#gof_4 <- model.gof(chr20_mtx, LF, 3)
#LF <- lfa(chr20_mtx, 10)
#gof_10 <- model.gof(chr20_mtx, LF, 3)
#hist(gof_4)
```



```{r}


data_list
```



```{r}
p_values_dt <-
  data_list  %>% 
  select(tidied) %>% 
  map_depth( 2, "p.value") %>% 
  map_depth( 2, c(2)) %>% # c(1) for the intercept
  unlist(use.names = FALSE) %>% 
  enframe(name = "locus" ,
          value= "p.value")%>%
  mutate(locus= data_list$locus)

p_values_dt
```


```{r}
estimates_dt <-
  data_list  %>% 
  select(tidied) %>% 
  map_depth( 2, "estimate")%>% 
  map_depth( 2, c(2)) %>% # c(1) for the intercept
  unlist(use.names = FALSE) %>% 
  enframe(name = "locus" ,
          value= "beta_hat")%>%
  mutate(locus= data_list$locus)

estimates_dt
```
```{r}
ggplot(p_values_dt)+
  geom_histogram((aes(x=p.value)), bins=100) +
  theme_bw()
```

get a p-value for target specific loci


```{r}

```

```{r}
p_values_subset <-
  data_list  %>% 
  filter(locus %in% target_loci) %>% 
  select(tidied) %>% 
  map_depth( 2, "p.value") %>% 
   map_depth( 2, c(2)) %>%
  unlist(use.names = FALSE) %>% 
  enframe(name = "locus" ,
          value= "p.value")%>%
  mutate(locus= target_loci)
p_values_subset 
```


```{r}

bas_dt <- bind_cols(bas_dt,
                    tibble(lf1=LF[,1], lf2=LF[,2],lf3=LF[,3]))
```


```{r}
ggplot(bas_dt,
       aes(lf2, lf3, color=Population)) +
  geom_point() +
  theme_bw()
```

### work with a subset

```{r}
chr20_sub <- chr20[,1:50000]
```

# PCA

compute the first three principal component vectors

```{r}
cm <- colMeans(chr20_sub)
p <- irlba(chr20_sub,
           nv=3,
           nu=3,
           tol=0.1,
           center=cm)
```


 add to dataset
```{r}
colnames(p$u) <- paste0("pc",1:3)
bas_dt <- bind_cols(bas_dt, 
                    as_tibble(p$u))
```



```{r}
ggplot(bas_dt,
       aes(pc1, pc2, color=Population)) +
  geom_point() +
  theme_bw() 
```







