---
title: "Legacy Data Confounds Modern Genomics Studies"
author: 
- name: Luke Anderson-Trocmé, Rick Farouni
  affiliation:
  - &cruk Department of Human Genetics, McGill University,  Montreal, Canada
date: '`r format(Sys.Date(), "%Y-%B-%d")`'
output:
  html_notebook:
    df_print: paged
    code_folding: show
    toc: yes
    toc_float: 
      collapsed: false
      smooth_scroll: false
---
  

load libraries
```{r, message = FALSE}
library(ggplot2)
library(RColorBrewer)
library(plotly)
library(colorspace)
library(cowplot)
library(data.table)
library(tidyverse)
library(purrr)
library(ggrepel)
library(OneR)
#Directories
DataPath <- '~/Documents/LegacyData/PlottingScripts/ProcessedData/'
figurePath <- '~/Documents/LegacyData/Figures/'

#Load the custom population colors
#Hex colors taken from https://www.nature.com/articles/nature15393
colors<-read.table(paste0(DataPath,
                          'NamePopBigPop_color.html.txt'), 
                   comment.char = '>',
                   col.names = c('Population',
                                 'hex',
                                 'BigPop'))

MyColour <- as.character(colors$hex)
names(MyColour) <- colors$Population

MyColour_tab <- as.character(colors$hex)
names(MyColour_tab) <- sub("^","                                       ",
                       colors$Population)
#extract legend
#https://github.com/hadley/ggplot2/wiki/Share-a-legend-between-two-ggplot2-graphs
g_legend <- function(a.gplot){
  tmp <- ggplot_gtable(ggplot_build(a.gplot))
  leg <- which(sapply(tmp$grobs, 
                      function(x) x$name) == "guide-box")
  legend <- tmp$grobs[[leg]]
  return(legend)}

```

# Figure 1
## Reverse GWAS in JPT
```{r,fig.height=5,fig.width=8}

SigPos <- readRDS(paste0(DataPath,
                   'SigPos.rds'))

NotSig <- readRDS(paste0(DataPath,
                   'NotSig.rds'))

facet_names <- 
  list('1'="1",'2'="",'3'="",'4'="",
       '5'="5",'6'="",'7'="",'8'="",'9'="",
       '10'="10",'11'="",'12'="",'13'="",'14'="",
       '15'="15",'16'="",'17'="",'18'="",'19'="",
       '20'="",'21'="",'22'="")

facet_labeller <- function(variable,value){
  return(facet_names[value])
}


#dummy plot to extract legend
dummy <- 
  ggplot(SigPos, 
         aes(x = Pos, 
             y = deviance, 
             shape = SigPos$Population)) +
  geom_point() +
  scale_shape_manual(values = c(16,16),
                     labels = c('Suspicious\nvariant')) +
  guides(shape = 
           guide_legend(title = '',
                        override.aes = 
                          list(size = 4,
                               color = c('royalblue1'))))+
  theme(legend.title = element_text(size = 20),
        legend.text = element_text(size = 20))

mylegend <- g_legend(dummy)

#make manhattan plot
gwas <- 
  ggplot(NotSig, aes(x = Pos, 
                     y = plog10, 
                     color = as.factor(Chr))) +
  geom_point(alpha = 0.3, 
             size = 0.5) +
  geom_point(data = SigPos, 
             aes(x = Pos, 
                 y = plog10),
             color = 'royalblue1',
             size = 0.5) +
  facet_grid(~Chr, 
             scales = 'free_x', 
             space = 'free_x', 
             switch = 'x',
             labeller = facet_labeller) +
  scale_y_continuous(expand = c(0,0)) +
  scale_color_manual(values = rep(
    c("grey70", "grey30"), 
    22)) +
  labs(title = "Association to the average quality of mapped bases in the 1kGP JPT",
       y = '-log10(p)', x = 'Chromosome') +
  guides(color = FALSE) + 
  theme_classic() +
  theme(plot.title = element_text(hjust = 0.5,
                                  size = 20), 
        plot.subtitle = element_text(hjust = 0.5), 
        axis.text.x = element_blank(), 
        axis.ticks.x = element_blank(),
        axis.line.x = element_blank(), 
        strip.background = element_blank(),
        strip.text = element_text(size = 20),
        axis.text = element_text(size = 20),
        axis.title.y = element_text(size = 20),
        axis.title.x = element_text(size = 20,
                                  vjust = 7),
        plot.margin = unit(c(1.5,1.5,0,1.5), "lines"))

GWAS <- plot_grid(gwas,
                  mylegend,
                  nrow = 1,
                  rel_widths = c(1,0.15))
```


## Joint Frequency Spectrum
```{r,fig.height=5,fig.width=5}
#load file if it already exists
#see MakeFiles.Rmd for info on jointAF.txt
join <- readRDS(paste0(DataPath,
               'jointAF.rds'))

#load poisitons that reach -log10(p) > 6
sig.6 <- SigPos[,c('Chr','Pos')]
sig.6$sig <- 'Significant' #add dummy variable for joining
join <- merge(join, 
              sig.6, 
              by=c('Chr', 'Pos'), 
              all.x=T)
join <- join[which(join$JPT_MAF + join$NAG_MAF != 0),] #remove fixed
jj <- join[complete.cases(join),]
```

```{r}
sig_points <- 
  ggplot() +  
  geom_bin2d(data=jj , 
             aes(x = NAG_MAF, 
                 y = JPT_MAF),
             binwidth = c(1/104, 1/104)) +
  scale_x_continuous(limits = c(-0.01,0.5)) +
  scale_y_continuous(limits = c(-0.01,0.5))

sig_points_data <- ggplot_build(sig_points)$data[[1]]

SFS <- 
  ggplot() + theme_classic() +
  geom_bin2d(data=join, 
             aes(x = NAG_MAF, 
                 y = JPT_MAF), 
             binwidth = c(1/104, 1/104)) +
  scale_fill_gradient(name = 'All variant\ncount',
                      low = "grey98",
                      high = "grey45",
                      trans = "log",
                      breaks = c(1,100,10000,1000000)) +
  geom_point(data = sig_points_data, 
             aes(x = x, 
                 y = y, 
                 size = count),
             fill = 'royalblue1',
             color = 'black',
             shape = 21) + 
  scale_size(name = 'Suspicious\nvariant\ncount',
             breaks = c(1, 10, 100), 
             range = c(1, 6)) +
  labs(x = "Nagahama alternate allele frequency",
       y = "1kGP JPT alternate allele frequency") +
  scale_x_continuous(limits = c(-0.01,0.5)) +
  scale_y_continuous(limits = c(-0.01,0.5)) +
  theme(axis.line = element_blank(),
        axis.text = element_text(size = 20),
        axis.title = element_text(size = 20),
        legend.text = element_text(size = 20),
        legend.title = element_text(size = 20),
        plot.margin = unit(c(0,0,1.5,1.5), "lines"))

#SFS
```




## Mutation Spectrum enrichment
```{r,fig.height=3,fig.width=9}
#load mutation spectrum for suspicious SNPs
SNP <- readRDS(paste0(DataPath,
                   'SNP_Context.rds'))
#load mutation spectrum for control SNPs
rSNP <- readRDS(paste0(DataPath,
                   'rSNP_Context.rds'))

#make color scale comparable between plots
maxFreq <- max(SNP$Freq)

plotContext <- function(data, name){
  
plt <- ggplot(data, 
              aes(x = End, 
                  y = Start, 
                  fill = Freq)) +
    geom_tile() +
    facet_grid(.~Mut) +
    labs(y = name) +
    theme_classic() +
    theme(axis.title.x = element_blank(),
          axis.line = element_blank(),
          axis.text = element_text(size = 20),
          axis.title.y = element_text(size = 20),
          strip.placement = "outside",
          strip.background = element_blank(),
          strip.text = element_text(size = 20),
          legend.text = element_text(size = 20),
          legend.title = element_text(size = 20),
          panel.border = element_rect(colour = "black", 
                                      fill = NA, 
                                      size = 1),
          plot.margin = unit(c(0,0,0,1.5), "lines")) +
    scale_fill_gradient(limits = c(0, maxFreq),
                        low = 'white', 
                        high = 'black',
                        guide = guide_legend(title = 'Variant\ncount',
                                             direction = 'vertical'))
return(plt)
}

SNPplt <- plotContext(SNP,'Suspicious')
rSNPplt <- plotContext(rSNP,'Control')


SNPplt <- SNPplt + 
  geom_point(data = 
               SNP[which(SNP$chi_p < 0.00001),], 
             aes(x=End, y = Start), 
             shape = '*', 
             color = 'royalblue1',
             size = 10) 


mylegend <- g_legend(rSNPplt)

Mut_Spect <- 
  plot_grid(
            plot_grid(
              SNPplt + 
                theme(legend.position = "none"),
              rSNPplt + 
                theme(legend.position = "none"),
              nrow = 2),
            mylegend,
            ncol = 2,
            rel_widths = c(1,0.15))

Mut_Spect


```
## Put all plots together and save
```{r,fig.height=11,fig.width=8}

fig1 <- plot_grid(GWAS,
                  SFS,
                  Mut_Spect,
                  ncol = 1,
                  rel_heights = c(2,6,2.5),
                  labels = c('A','B','C'),
                  label_size = 20)

ggsave(paste0(figurePath,
              'Figure1.jpg'),
       fig1, 
       height = 21.25,
       width = 14)

```


```{r}
#save memory
rm(GWAS)
rm(fig1)
rm(A)
rm(B)
rm(AB)
```

#Figure 2
This plot illustrates the quality of the date from the 1000 Genomes Project over time.
Before we can plot this, we must download and parse the data.
See ProcessPlotData.Rmd for info

### Quality per population
```{r,fig.height=10,fig.width=8}
metaData <- readRDS(paste0(DataPath,
                           'metaData.rds'))

last_ind <- metaData %>% 
             group_by(Population) %>%
             filter(avg_qual == max(avg_qual))
last_ind <- last_ind[!duplicated(last_ind[,c('Population')]),]
m <- names(metaData)

last_ind$text <- last_ind$Population

metaData <- merge(metaData,last_ind, by = m, all = T)
metaData[is.na(metaData)] <- ""

a <- 
  ggplot(metaData, 
         aes(x = sample,
             y = avg_qual,
             color = Population,
             shape = as.factor(Phase))) +
  geom_point(size=1) +
  geom_text_repel(segment.size = 0,
                  #min.segment.length = 1,
                  aes(label = text))+
  geom_hline(yintercept = 30,
             linetype = 2,
             color = 'grey') +
  scale_colour_manual(values = MyColour, 
                      breaks = unique(metaData$Population)) +
  scale_x_discrete(expand = c(0.1,0.1)) +
  scale_shape_manual(values = c(1,3),
                     name = 'Phase') +
  guides(color = FALSE) +
  labs(x = "Populations ranked by sequencing date",
       y = "Average quality of mapped bases") +
  theme(legend.position = c(0.9, 0.95),
        axis.text.x = element_blank(), 
        axis.ticks.x = element_blank(),
        panel.spacing = unit(0, "lines"),
        strip.placement = 'inside',
        strip.background = element_blank())

a
```

### Sequencing technology use over time
```{r,fig.height=2,fig.width=6}
c <- 
  ggplot(metaData, 
         aes(x = SUBMISSION_DATE, 
             y = INSTRUMENT_MODEL,
             shape = INSTRUMENT_MODEL)) +
  geom_point(size=1) +
  labs(x = "Sequencing date", 
       y = 'Sequencer') +
  guides(shape = guide_legend(title = "",
                              label = FALSE,
                              override.aes = list(alpha = 0))) +
  scale_shape_manual(values = c(2,4,5,6)) +
  scale_x_date(date_labels = "%Y",
               date_breaks = "1 year") +
  theme(axis.line.y = element_blank(),
        axis.text.y = element_text(size = 10,
                                   hjust = 0),
        plot.margin = unit(c(0,0,0,1.5), "lines"))

c

```

Put it all together and save it
```{r,fig.height=11,fig.width=8}
d <- plot_grid(a,
               c, 
            nrow = 2, 
            labels = c('A', 'B'),
            rel_heights = c(5,1),
            align = 'h')
ggsave(paste0(figurePath,
              "MapQualOverTime.jpg"),
       d,
       height = 9,
       width = 9)

d
```

### Quality by sequencing center
```{r,fig.height=5,fig.width=5}
b <- ggplot(metaData, 
            aes(x = SUBMISSION_DATE, 
                y = avg_qual, 
                color = CENTER_NAME)) +
            geom_point(shape = 0) + 
            geom_hline(yintercept = 30, 
                       linetype = 2, 
                       color = 'grey') +
            scale_color_brewer(palette = 'RdYlBu') +
            guides(color = guide_legend(title = "Sequencing\n    Center",
                                        override.aes = list(shape = 15,
                                                            size = 5))) +
            labs(y = "Average quality of mapped bases",
                 x = 'Sequencing Date')
b

ggsave(paste0(figurePath,
              "quality_by_center.jpg"),
       b,
       height = 9,
       width = 9)

```

#Figure 3

This figure shows the overlap of sites associated to qualiy independantly in each single population regression.
See ProcessPlotData.Rmd for info on the overLap file

### Overlap of significant SNPs
```{r,fig.height=5,fig.width=5}
overLap <- readRDS(paste0(DataPath,
                           'overLap.rds'))

#remove variants that don't pass strict mask
overLap <- overLap %>% dplyr::filter(strict == "pass")

pop_order_single <- overLap %>% arrange(-single_count) %>% select(Population) %>% unique() %>% unlist()

pop_order_double <- overLap %>% arrange(-double_count) %>% select(Population) %>% unique() %>% unlist()

overLap$Population <- factor(overLap$Population, levels = pop_order_single)
p1.1 <- ggplot(overLap, 
       aes(x=Population, 
           y= single_count)) + 
  geom_bar(stat="identity") +
  scale_fill_grey() +
  theme(axis.text.x = element_text(angle = 60, hjust = 1, size = 20),
        axis.text.y = element_text(size = 20),
        axis.title = element_text(size = 20),
        plot.title = element_text(size = 20)) +
  labs(x = "Population", y = "Count",
       title = "Q-associated in only one population")

overLap$Population <- factor(overLap$Population, levels = pop_order_double)
p1.2 <- ggplot(overLap, 
       aes(x=Population, 
           y= double_count)) + 
  geom_bar(stat="identity") +
  scale_fill_grey() +
  theme(axis.text.x = element_text(angle = 60, hjust = 1, size = 20),
        axis.text.y = element_text(size = 20),
        axis.title = element_text(size = 20),
        plot.title = element_text(size = 20)) +
  labs(x = "Population", y = "Count",
       title = "Q-associated in at least two population")

p1 <- plot_grid(p1.1,p1.2)

p1
```
```{r}
all_sites <- all_sites %>% dplyr::filter(strict =="pass")

pop_order <- all_sites %>% 
  arrange(-total_count) %>% 
  select(Population) %>% 
  unique() %>% 
  unlist()

all_sites$Population <- factor(all_sites$Population, levels = pop_order)
p1.a <- ggplot(all_sites, 
       aes(x=Population, 
           y= total_count)) + 
  geom_bar(stat="identity") +
  theme(axis.text.x = element_text(angle = 60, hjust = 1, size = 20),
        axis.text.y = element_text(size = 20),
        axis.title = element_text(size = 20),
        plot.title = element_text(size = 20)) +
  labs(x = "Population", y = "Variant count",
       title = "Q-associated in single population test")

p1.a
```
### Frequency of occurence
```{r,fig.height=2,fig.width=5}
#save to file
pos_count <- readRDS(paste0(DataPath,
                   'pos_count.rds'))
pos_count <- pos_count %>% dplyr::filter(strict =="pass")
#pos_count$strict <- paste(pos_count$strict, "strict mask") 
p2 <- 
  ggplot(pos_count, aes(x = count, y = freq)) +
  geom_bar(stat= "identity") +
  scale_x_continuous(breaks = seq(1,13)) +
  labs(x = "Number of populations with Q-associations to the same variant",
       y = "Count") +
  theme(axis.text = element_text(size = 20),
        axis.title = element_text(size = 20),
        strip.text = element_text(size = 20))

p2
```

put it all together and save it
```{r,fig.height=11,fig.width=8}
p3 <- plot_grid(p1.a, p2, 
                ncol = 1, 
                align = "v", 
                rel_heights = c(3, 1), 
                labels = c("A","B"))

ggsave(paste0(figurePath,
              'SNPOverlap6.jpg'),
              p3, 
              height = 12, 
              width = 12)

p3
```


#Supplementary Figures

## Mutation Spectrum enrichment of GCAT SNPs
```{r,fig.height=3,fig.width=9}
#load mutation spectrum for suspicious SNPs
SNP <- readRDS(paste0(DataPath,
                   'strict.gcat.SNP_Context.rds'))
#load mutation spectrum for control SNPs
rSNP <- readRDS(paste0(DataPath,
                   'strict.gcat.rSNP_Context.rds'))

#make color scale comparable between plots
maxFreq <- max(SNP$Freq)

plotContext <- function(data, name){
  
plt <- ggplot(data, 
              aes(x = End, 
                  y = Start, 
                  fill = Freq)) +
    geom_tile() +
    facet_grid(.~Mut) +
    labs(y = name) +
    theme_classic() +
    theme(axis.title.x = element_blank(),
          axis.line = element_blank(),
          axis.text = element_text(size = 20),
          axis.title.y = element_text(size = 20),
          strip.placement = "outside",
          strip.background = element_blank(),
          strip.text = element_text(size = 20),
          legend.text = element_text(size = 20),
          legend.title = element_text(size = 20),
          panel.border = element_rect(colour = "black", 
                                      fill = NA, 
                                      size = 1),
          plot.margin = unit(c(0,0,0,1.5), "lines")) +
    scale_fill_gradient(limits = c(0, maxFreq),
                        low = 'white', 
                        high = 'black',
                        guide = guide_legend(title = 'count',
                                             direction = 'vertical'))
return(plt)
}

SNPplt <- plotContext(SNP,'Suspicious')
rSNPplt <- plotContext(rSNP,'Control')
chisq_min <- 5

SNPplt <- SNPplt + 
  geom_point(data = 
               SNP[which(SNP$chi_p < 0.00001),], 
             aes(x=End, y = Start), 
             shape = '*', 
             color = 'royalblue1',
             size = 10) 


mylegend <- g_legend(rSNPplt)

Mut_Spect_gcat <- 
  plot_grid(
            plot_grid(
              SNPplt + 
                theme(legend.position = "none"),
              rSNPplt + 
                theme(legend.position = "none"),
              nrow = 2),
            mylegend,
            ncol = 2,
            rel_widths = c(1,0.15))

Mut_Spect_gcat

ggsave(paste0(figurePath,
              'strict_gcat_mutational_enrichment.jpg'),
              height=4, 
              width=10)
```


```{r}
freq_gw <- data.frame('freq' = join$JPT_MAF,
                      'SNPs' = 'All')
freq_sig <- data.frame('freq' = jj$JPT_MAF,
                       'SNPs' = 'Suspicious')

freq_hist <- rbind(freq_sig, freq_gw)

HIST <- 
  ggplot(data = freq_hist %>%
           filter(freq > 0),
         aes(x=freq, 
             fill = SNPs))+
  geom_histogram(bins = 100,
                 alpha=0.8, 
                 position="identity",
                 aes(y=..density..))+
  geom_hline(yintercept = 16.5, 
             color = 'white',
             size = 5) +
  coord_cartesian(ylim = c(0, 18)) +
  scale_fill_manual(values=c("royalblue1", "grey30")) +
  scale_x_continuous(expand = c(0.01, 0.01)) +
  scale_y_continuous(expand = c(0.01, 0.01), 
                  breaks = c(0,5,10,15,16.5,18),
                  labels = c('0','5','10','15','...','30'))+
  labs(y="Density",
       x = "1kGP Allele Frequency") +
  theme_classic() +
  theme(axis.line = element_blank(),
        axis.ticks = element_blank(),
        axis.text = element_text(size = 20),
        axis.title = element_text(size = 20),
        legend.title = element_text(size = 20),
        legend.text = element_text(size = 20))

HIST

ggsave(paste0(figurePath,
              "histogram_of_sig_snps.jpg"),
       HIST,
       height = 6,
       width = 9)

```

##Comparig models
```{r,fig.height=11,fig.width=8}
fitsName = 'fits_Significant_Positions_'
#load data
LFA <- fread(paste0(DataPath,fitsName,"LFA.txt"))
  
Pop <- fread(paste0(DataPath,fitsName,"Pop.txt"))
  
PCA <- fread(paste0(DataPath,fitsName,"PCA.txt"))

plt <- merge(LFA,Pop,by=c('rsID'),suffixes = c('LFA','PCA'))
plt <- merge(plt,PCA,by=c('rsID'))

p1a<-ggplot(plt, aes(b, bPCA)) +
  geom_abline(color='grey') +
  geom_point(shape=1, alpha=0.2) +
  labs(x='Pop', y='PCA')

p1b<-ggplot(plt, aes(b, bLFA)) +
  geom_abline(color='grey') +
  geom_point(shape=1, alpha=0.2) +
  labs(x='Pop', y='LFA') +
  ggtitle('Beta')

p1c<-ggplot(plt, 
            aes(bLFA, 
                bPCA)) +
    geom_abline(color='grey') +
  geom_point(shape=1, alpha=0.2) +
  labs(x='LFA', y='PCA')

p2a<-ggplot(plt, 
            aes(devdiff, 
                devdiffPCA))+
  geom_abline(color='grey')+
  geom_point(shape=1, alpha=0.2)+
  labs(x='Pop', y='PCA')
p2b<-ggplot(plt, 
            aes(devdiff, 
                devdiffLFA))+
  geom_abline(color='grey')+
  geom_point(shape=1, alpha=0.2)+
  labs(x='Pop', y='LFA')+
  ggtitle('Deviance')
p2c<-ggplot(plt, 
            aes(devdiffLFA, 
                devdiffPCA))+
  geom_abline(color='grey')+
  geom_point(shape=1, alpha=0.2)+
  labs(x='LFA', y='PCA')

plot_grid(p1b,p2b,
          p1a,p2a,
          p1c,p2c,
          nrow=3)

ggsave(paste0(figurePath,fitsName,
              'CompareModels.jpg'),
              height=10, 
              width=8)
```


## Comparing 1kGP to Resequenced Han
###Single Population Test
```{r,fig.height=5,fig.width=5}
#using the sites associated to Quality in the single population test
HanKgpAF <- readRDS(paste0(DataPath,
                   'HanKgpAF.rds'))

ggplot(HanKgpAF, 
       aes(x = hanAF/2, 
           y = kgpAF/2))+
  geom_bin2d(bins = 83)+
  scale_fill_distiller(palette = "Spectral")+
  labs(x = 'Resequenced Han', 
       y = '1000 Genomes Project')+
  ggtitle(paste0('Joint frequency plot of ',nrow(HanKgpAF),' suspicious SNPs\nfrom single population test'))

ggsave(paste0(figurePath,'Han_1kGP_SFS_singlePop.jpg'),height=10, width=10)

```
##Full Model
```{r,fig.height=5,fig.width=5}
#Load joint frequency for SNPs
##from plink --freq
kgpAF <- fread(paste0(DataPath,'Significant_Positions.frq'))
hanAF <- fread(paste0(DataPath,'Significant_Positions_Han.frq'))
combo <- merge(hanAF, kgpAF, 
               by=c('CHR','SNP'), 
               suffixes = c('.HAN','.KGP'), 
               all = T)
combo[is.na(combo)] <- 0
#keep only sites present in 83 KGP samples
combo <- combo[which(combo$MAF.KGP > 0),]
combo$diff <- abs(combo$MAF.KGP - combo$MAF.HAN)
nHan <- nrow(combo[which(combo$MAF.HAN > 0),])

ggplot(combo, 
        aes(x = MAF.HAN, 
            y = MAF.KGP)) +
     geom_bin2d(bins = 100) +
     scale_fill_distiller(palette = "Spectral")+
    labs(x = 'Resequenced Han', 
         y = '1000 Genomes Project')+
    ggtitle(paste0('Joint frequency spectrum of \n',nrow(combo),' suspicious variants from combined test\n(',nHan, ' variants present in Han)'))

ggsave(paste0(figurePath,'Han_1kGP_SFS_FullModel.jpg'),height=10, width=10)


```

##NYGC Frequencies
```{bash}
for f in `seq 1 22`; \
do vcftools \
--gzvcf ./CCDG_13607_B01_GRM_WGS_2019-02-19_chr${f}.recalibrated_variants.vcf.gz \
--freq --out ./chr${f}; \
done
```

##NYGC ReSequence Data
```{r,fig.height=5,fig.width=5}
#Load joint frequency for SNPs
##from plink --freq
kgpAF <- fread('/Users/luke/genomes/genomes/hg19/Freq/genomeWide_1kGP.frq',
               col.names = c('Chr','Pos','N_Alleles','N_Counts','AF1','AF2'))
kgpAF <- kgpAF %>%
  separate(AF1, c("Allele1", "AF1"), ":")
kgpAF <- kgpAF %>%
  separate(AF2, c("Allele2", "AF2"), ":")
nygcAF <- fread(paste0(DataPath,'NYGC/genomeWide_NYGC_lifted_1-9.frq'),
               col.names = c('Chr','Pos','PosPlus1','N_Counts','AF1','Af2'))
nygcAF <- nygcAF %>%
  separate(AF1, c("Allele1", "AF1"), ":")
nygcAF$Chr <- as.numeric(as.character(gsub(pattern = 'chr',replacement = '',x = nygcAF$Chr)))

sigPos <- fread(paste0(DataPath,'Significant_Positions.txt'), 
                col.names = c('Chr','Pos'))

strictMask <- fread(paste0(DataPath,
                           '1kGP_StrictMask.bed'),
            col.names = c('Chr','Pos','Pos.1','rsID'))

combo <- merge(nygcAF, kgpAF, 
               by=c('Chr','Pos'),
               suffixes = c('.NYGC','.KGP'), 
               all = T)

combo <- combo[which(combo$Allele1.KGP == combo$Allele1.NYGC),]

combo$AF1.NYGC <- as.numeric(as.character(combo$AF1.NYGC))
combo$AF1.KGP <- as.numeric(as.character(combo$AF1.KGP))

combo[is.na(combo)] <- 0
combo$diff <- abs(combo$AF1.KGP - combo$AF1.NYGC)


sig <- merge(combo, sigPos, by=c('Chr','Pos'))

combo <- anti_join(combo,
                   strictMask,
                   by = c('Chr','Pos'))

combo <- combo[which((combo$AF1.NYGC > 0) &
                       (combo$AF1.KGP > 0)),]
combo <- combo[which((combo$AF1.NYGC < 1) &
                       (combo$AF1.KGP < 1)),]

sfs <- ggplot(sig, 
        aes(x = 1-AF1.NYGC, 
            y = 1-AF1.KGP)) +
     geom_bin2d(bins=250) +
     scale_fill_distiller(palette = "Spectral",
                          trans = "log",
                          breaks = c(5,50,500,5000))+
     scale_y_continuous(limits = c(0,1),
                        expand = c(0.01, 0.01)) +
    labs(x = 'Resequenced NYGC', 
         y = '1000 Genomes Project') +
    ggtitle('Suspicious Variants in Chromosome 1-9')

sfs

ggsave(paste0(figurePath,'NYGC_Chr1-9.jpg'),height=10, width=10)

allsfs <- ggplot(combo, 
        aes(x = 1-AF1.NYGC, 
            y = 1-AF1.KGP)) +
     geom_bin2d(bins=250) +
     scale_fill_distiller(palette = "Spectral",
                          trans = "log",
                          breaks = c(1,10,100,1000,10000,100000,1000000))+
     scale_y_continuous(limits = c(0,1),
                        expand = c(0.01, 0.01)) +
    labs(x = 'Resequenced NYGC', 
         y = '1000 Genomes Project') +
    ggtitle('All Variants in Chromosome 1-9')



allsfs

ggsave(paste0(figurePath,'NYGC_Chr1-9_all.jpg'),height=10, width=10)

sfsPlus <- ggplot(combo, 
        aes(x = 1-AF1.NYGC, 
            y = 1-AF1.KGP)) +
     geom_bin2d(bins=250) +
     geom_point(data = sig, size = 0.8, alpha = 0.7, shape = 3) +
     scale_fill_distiller(palette = "Spectral",
                          trans = "log",
                          breaks = c(1,10,100,1000,10000,100000,1000000))+
     scale_y_continuous(limits = c(0,1),
                        expand = c(0.01, 0.01)) +
    labs(x = 'Resequenced NYGC', 
         y = '1000 Genomes Project') +
    ggtitle('All Variants in Chromosome 1-9')



sfsPlus

ggsave(paste0(figurePath,'NYGC_Chr1-9_allPlus.jpg'),height=10, width=10)
```
```{r}
hist <- ggplot(sig, aes(x=diff)) +
            geom_histogram(binwidth = 0.01) +
            scale_x_continuous(limits = c(0,1),
                             expand = c(0.01, 0.05)) +
            scale_y_reverse(limits=c(3000,0)) +
            labs(x = "Difference in Frequency",
                 y = "") +
            coord_flip() +
            theme_classic() +
            theme(axis.line = element_blank())

```

```{r,fig.height=11,fig.width=8}
AF30 <- readRDS(paste0(DataPath,
                   'AF30.rds'))
AF30$rate <- AF30$oAF/AF30$uAF

af1 <- ggplot(AF30, aes(x=uAF, y=oAF))+
  geom_bin2d(bins=200)+
  scale_fill_distiller(palette = "Spectral")+
  geom_abline(intercept=0, slope = 2279/225, linetype=2, color='black')+
  geom_abline(intercept=0, slope = 2279/450, linetype=3, color='red')+
  labs(x='Samples with Q < 30',y='Samples with Q > 30')

af2 <- ggplot(AF30[which(AF30$oAF <100),], aes(x=uAF, y=oAF))+
  geom_bin2d(bins=100)+
  scale_fill_distiller(palette = "Spectral")+
  geom_abline(intercept=0, slope = 2279/225, linetype=2, color='black')+
  geom_abline(intercept=0, slope = 2279/450, linetype=3, color='red')+
  labs(x='Samples with Q < 30',y='Samples with Q > 30')

plot_grid(af1, af2,ncol=1)

ggsave(paste0(figurePath,'OverUnder30.jpg'),height=11,width=8)
ggsave(paste0(figurePath,'OverUnder30.pdf'),height=11,width=8)

print(nrow(AF30[which(AF30$rate <= 2279/450),])/nrow(AF30))
print(nrow(AF30[which(AF30$rate > 2279/450),])/nrow(AF30))
```
##Over Under 30 for published variants
```{r}
pub<-fread(paste0(DataPath, 'fits_PublishedVariants_LFA.txt'))
pub <- merge(pub, AF30, by='rsID')
pub$text <- paste(pub$rsID, 'p ',round(pub$dev_logp, 2))
library(ggrepel)
ggplot(pub, aes(x=uAF, y=oAF, label=text))+
     geom_point()+
     geom_text_repel(size=3)+
     geom_abline(intercept=0, slope = 2279/225, linetype=2, color='black')+
     geom_abline(intercept=0, slope = 2279/450, linetype=3, color='red')+
     labs(x='Samples with Q < 30',y='Samples with Q > 30')+
     ggtitle('Published Variants')
ggsave(paste0(figurePath,'Published_OverUnder30.jpg'),height=7,width=8)
pub$ratio <- pub$oAF/pub$uAF
pub$limit <- pub$ratio > 2279/450
write.table(pub,paste0(figurePath,'Published_OverUnder30.txt'),)

```

```{r,fig.height=5,fig.width=10}
#load file
omni <- readRDS(paste0(DataPath,'omni.rds'))

ggplot(omni, 
       aes(x = (2504 - num_zeros)/2504, 
           y = dev_logpLFA)) +
  geom_point(shape = 1) +
  labs(x = 'Global Allele Frequency',
       y = '-log10(p)') +
  ggtitle(paste0(nrow(omni),' SNPs present in the Omni 2.5 Microarray'))

ggsave(paste0(figurePath,'Omni_AF.jpg'), 
       height = 5, 
       width = 10)

```

##per chromosome
```{r}
#get number of sig positions per chromosome
sigPerChrom <- 
  as.data.frame(table(
    RegressionMasked[which(
      RegressionMasked$Adjusted_Log10p > -log10(0.01)),]$Chr))
#get number of sig positions per chromosome
PerChrom <- as.data.frame(table(RegressionMasked$Chr))
sigPerChrom <- merge(sigPerChrom, PerChrom, by='Var1')
sigPerChrom$density <- sigPerChrom$Freq.x/sigPerChrom$Freq.y


#for each chrom
snp_list <- list()
snp_combined <- data.frame()
maxFreq <- 0
for(chrom in 1:22) {
  snp_list[[chrom]] <- 
    processContext(SNP %>% filter(Chr == chrom)) #get context of each sig var
  snp_list[[chrom]]$Chr <- chrom #add chrom number for merging
  snp_combined <- rbind(snp_combined,snp_list[[chrom]]) #row bind
}

#modify title for plotting
snp_combined$title <- 
  paste0(snp_combined$Start,
         substr(snp_combined$Mut,1,1),
         snp_combined$End,
         substr(snp_combined$Mut,2,5))

#get chromosome length
ChromLength <- fread(paste0(DataPath,
                            'GRCh37_ChromLength.txt'),
                     sep='\t',
                     col.names = c('Chr','Length'))
snp_combined <- merge(snp_combined, 
                      ChromLength, 
                      by = 'Chr')
snp_combined <- merge(snp_combined, 
                     PerChrom, 
                     by.x='Chr',
                     by.y='Var1')
#normalize by length
snp_combined$len_norm_freq <- snp_combined$Freq / snp_combined$Length
#normalize by number of tests
snp_combined$n_norm_freq <- snp_combined$Freq.x / snp_combined$Freq.y

signature <- c("TAT → T", "TCT → T", "ACA → A")

snp_combined$Signature = 'other'
snp_combined[grepl("AC → C",
                   snp_combined$title),]$Signature <- '*AC→*CC'
snp_combined[grepl(paste(signature,
                         collapse="|"),
                   snp_combined$title),]$Signature <- 'T*T→TTT'

snp_combined$Signature <- factor(snp_combined$Signature, 
                                 levels = c("*AC→*CC", "T*T→TTT", "other"))
mut_count_chrom <-
  ggplot(snp_combined, 
         aes(x=as.factor(Chr), 
             y = n_norm_freq,
         color = Signature))+
  geom_point(shape = 1,
             size = 4) +
  scale_color_manual(values = c('blue','red','black'))+
  geom_text_repel(data = snp_combined %>% 
                    filter(n_norm_freq > 0.00025),
                  show.legend = FALSE,
                  aes(label = title),
                  size = 3,
                  force = 2)+
  labs(y = 'proportion of significant variants',
       x = 'Chromosome')+
   theme(axis.text=element_text(size=20),
         axis.title=element_text(size=20),
         legend.text=element_text(size=20),
         legend.title=element_text(size=20),
         legend.position = c(0.9, 0.8))


ggsave(paste0(figurePath,
              'mutational_enrichment_per_chrom.jpg'),
              height=5, 
              width=12) 
#reject
geom_point(data = filter(snp_combined, 
                           grepl("AC → C",title)),
             aes(x=as.factor(Chr), 
             y = n_norm_freq),
             color = 'royalblue1',
             shape = 1) +
  geom_point(data = filter(snp_combined, 
                           grepl(paste(signature,
                                       collapse="|"),
                                 title)),
             aes(x=as.factor(Chr), 
             y = n_norm_freq),
             color = 'red',
             shape = 1) +
sig_per_chrom <-
  ggplot(sigPerChrom, 
         aes(x=Var1,y=density))+
  geom_bar(stat="identity")+
  labs(x='Chromosome',
       y='proportion of significant variants')+
   theme(axis.text=element_text(size=20),
         axis.title=element_text(size=20))

per_chr <-
  plot_grid(sig_per_chrom, 
          mut_count_chrom, 
          labels = 'AUTO', 
          ncol =1, 
          align = 'v')
ggsave(paste0(figurePath,
                'CountPerChrom.jpg'), 
       height=8, 
       width=11)
```
##Density of Significant SNPs per Chromosome
```{r}
ChromLength <- fread(paste0(DataPath,
                            'GRCh37_ChromLength.txt'),
                     sep='\t',
                     col.names = c('Chr','Length'))
plts <- list()
bins <- list()
bins_mask <- list()
for(chrom in seq(1,22,1)){
  print(chrom)
  df <- RegressionMasked[which((RegressionMasked$Chr == chrom) &
                           (RegressionMasked$Adjusted_Log10p 
                            > -log10(0.01))),
                   c('Chr','Pos')]
  mask <- Regression[which((Regression$Chr == chrom) &
                           (Regression$Adjusted_Log10p 
                            > -log10(0.01)) &
                             !(Regression$rsID %in% RegressionMasked$rsID)),
                   c('Chr','Pos')]
  
  l = ChromLength[which(ChromLength$Chr == chrom),
                  c('Length')][[1]]
  b = seq(1,l,by=1000000)
  if(chrom == 1){
    plts[[chrom]]<-ggplot(mask, aes(x=Pos)) +
    geom_freqpoly(breaks = b, color = 'grey60') +
    geom_freqpoly(data = df, breaks = b) +
    ylim(0,50)+
    theme(axis.text.x = element_blank(),
          axis.ticks.x = element_blank(),
          axis.line = element_blank(),
          plot.margin = margin(0,0,0,10),
          axis.text=element_text(size=20),
          axis.title=element_text(size=20)) + 
    labs(x=chrom,
         y='Significant variants per 1Mb window')
  }
  else if(chrom == 22){
    plts[[chrom]]<-ggplot(mask, aes(x=Pos)) +
    geom_freqpoly(breaks = b, color = 'grey60') +
    geom_freqpoly(data = df, breaks = b) +
    ylim(0,50)+
    theme(axis.text = element_blank(),
          axis.ticks = element_blank(),
          axis.title.y = element_blank(),
          axis.line = element_blank(),
          plot.margin = margin(0,1,0,0),
          axis.title=element_text(size=20)) + 
    xlab(chrom)
  }
  else{
  plts[[chrom]]<-ggplot(mask, aes(x=Pos)) +
    geom_freqpoly(breaks = b, color = 'grey60') +
    geom_freqpoly(data = df, breaks = b) +
    ylim(0,50)+
    theme(axis.text = element_blank(),
          axis.ticks = element_blank(),
          axis.title.y = element_blank(),
          axis.line = element_blank(),
          plot.margin = margin(0,0,0,0),
          axis.title=element_text(size=20)) + 
    xlab(chrom)
  }
  #get freq count per bin
  c <- cut(df$Pos, breaks = b)
  bins[[chrom]] <- as.data.frame(table(c))
  
  c <- cut(mask$Pos, breaks = b)
  bins_mask[[chrom]] <- as.data.frame(table(c))
}

 TotalBinCount <- 
   bins %>% 
   bind_rows(.id = "source") %>%
   group_by(Freq) %>%
   summarise(Count = n())
 
  TotalBinCount_mask <- 
   bins_mask %>% 
   bind_rows(.id = "source") %>%
   group_by(Freq) %>%
   summarise(Count = n()) 
 
  TotalBinCount_mask$mask <- "Fail"
  TotalBinCount$mask <- "Pass"
  BinCount <- rbind(TotalBinCount, TotalBinCount_mask)
 ppHist <- 
   ggplot(BinCount,
          aes(x = Freq, y = Count, color = mask)) +
   geom_point(shape = 1,
              size = 4) +
   scale_y_log10()+
   scale_color_manual(values = c("grey60", "black")) +
   guides(color=guide_legend(title="Strict mask")) +
   theme_classic() +
   labs(x='Number of Q-associated variants in a 1mb window',
        y='-log10(count)')+
   theme(axis.text=element_text(size=20),
         axis.title=element_text(size=20),
         legend.text=element_text(size=20),
         legend.title=element_text(size=20),
         legend.position = c(0.95, 0.8))

  
relWidths <- unlist(ChromLength$Length)/100000000
relWidths[[1]] <- relWidths[[1]]*1.5
p1000 <- plot_grid(plts[[1]],plts[[2]],plts[[3]],plts[[4]],plts[[5]],plts[[6]],plts[[7]],plts[[8]],plts[[9]],plts[[10]],plts[[11]],plts[[12]],plts[[13]],plts[[14]],plts[[15]],plts[[16]],plts[[17]],plts[[18]],plts[[19]],plts[[20]],plts[[21]],plts[[22]],nrow=1,rel_widths = relWidths)

plot_grid(sig_per_chrom, 
          mut_count_chrom, 
          p1000,
          ppHist,
          ncol=1,
          labels = "AUTO")

ggsave(paste0(figurePath,
                'PerChromosome.jpg'), 
       height=20, 
       width=20)


p2 <- plot_grid(NULL,
                p, 
                ppHist,
                nrow=3, 
                rel_heights = c(1,5,5),
                labels = c("A","","B"))

ggsave(paste0(figurePath,
                'VariantDensity.jpg'), 
       height=5, 
       width=10)

#get 10kb regions with more than 10 sig snps
over10  <- bins %>%
   bind_rows(.id = "source") %>%
   filter(Freq >= 5) %>%
   separate(c, c("start_bp", "end_bp"), ",")

over10$start_bp <- as.numeric(as.character(substring(over10$start_bp,2)))
over10$end_bp <- as.numeric(as.character(substring(over10$end_bp,1,
                                                    nchar(over10$end_bp)-1)))
names(over10)[c(1,4)]<-c('Chr','Freq_Sig_Var')
write.table(over10,
            file = paste0(DataPath,
                          'RegionsOver10.bed'),
            row.names = FALSE, 
            quote = FALSE,
            sep = '\t')

 makeSubPlot <- function(df,Name){
p<-ggplot(df, 
       aes(x=Pos, 
           y = Adjusted_Log10p))+
  geom_point(size = 1) +
  expand_limits(y=c(0,max(df$Adjusted_Log10p)+1))  +
  geom_hline(yintercept = -log10(0.01), 
             color = 'blue') +
  labs(y = '-log10(p)', 
       title = Name) +
  theme_classic() +
  theme(plot.title = element_text(hjust = 0.5), 
        axis.text.x = element_blank(), 
        axis.ticks.x = element_blank())

return(p)
}

top10<-head(over10[order(-over10$Freq),],10)
 
 sub <- list()
 pp <- list()
 for(i in seq(1,nrow(top10))){
   chr <- top10[i,]$Chr
   start <- top10[i,]$start_bp
   end <- top10[i,]$end_bp
   title <- paste0('Chr',chr, ':',start,'-',end)
   sub[[i]] <- RegressionMasked[which((RegressionMasked$Chr == chr) &
                              (RegressionMasked$Pos > start) &
                              (RegressionMasked$Pos < end)),
                              c('Chr','Pos','rsID',
                                'Adjusted_Log10p',
                                'Repeat','Indel')]
   pp[[i]] <- makeSubPlot(sub[[i]],title)
 }

 pp[[1]]
  ggsave(paste0(figurePath,
                'RegionOver10.jpg'), 
       height=5, 
       width=8)
  
 ggsave(paste0(figurePath,
                'top10.jpg'), 
       height=12, 
       width=8)
 
 
 library(Homo.sapiens)
 library(annotate)
 mycoords.gr = over10 %>%
   dplyr::select(chrom=Chr, start=start_bp, end=end_bp) %>%
   mutate(chrom=paste0('chr', chrom)) %>%
   makeGRangesFromDataFrame
 
 GeneRegions <- subsetByOverlaps(genes(TxDb.Hsapiens.UCSC.hg19.knownGene), mycoords.gr)
 
out <- data.frame(seqnames=seqnames(GeneRegions),
  starts=start(GeneRegions)-1,
  ends=end(GeneRegions),
  names=c(rep(".", length(GeneRegions))),
  scores=c(rep(".", length(GeneRegions))),
  strands=strand(GeneRegions),
  gene_id=GeneRegions$gene_id)

out$gene_symbol <- getSYMBOL(as.character(unlist(out$gene_id)), 
                        data='org.Hs.eg')

write.table(out, file = paste0(DataPath,
                          'RegionsOver10_geneID.bed'), 
                          quote=F, 
                          sep="\t", 
                          row.names=F)
```


```{r}
sessionInfo()
```

