---
title: "Legacy Data Confounds Modern Genomics Studies"
author: 
- name: Luke Anderson-Trocm√©, Rick Farouni
  affiliation:
  - &cruk Department of Human Genetics, McGill University,  Montreal, Canada
date: '`r format(Sys.Date(), "%Y-%B-%d")`'
output:
  html_notebook:
    df_print: paged
    code_folding: show
    toc: yes
    toc_float: 
      collapsed: false
      smooth_scroll: false
---
  

load libraries
```{r, message = FALSE}
library(ggplot2)
library(RColorBrewer)
library(plotly)
library(colorspace)
library(cowplot)
library(data.table)
library(tidyverse)
library(purrr)
library(ggrepel)
library(OneR)
#Directories
project_dir <- getwd()
data_dir <- file.path(project_dir, "ProcessedData/")
figure_dir <- file.path(project_dir, "Figures/")

#data_dir <- '~/Documents/LegacyData/PlottingScripts/ProcessedData/'
#figure_dir <- '~/Documents/LegacyData/Figures/'

#Load the custom population colors
#Hex colors taken from https://www.nature.com/articles/nature15393
colors<-read.table(paste0(data_dir,
                          'NamePopBigPop_color.html.txt'), 
                   comment.char = '>',
                   col.names = c('Population',
                                 'hex',
                                 'BigPop'))
MyColour <- as.character(colors$hex)
names(MyColour) <- colors$Population
```

# Figure 1
## Reverse GWAS in JPT
```{r,fig.height=5,fig.width=8}

SigPos <- readRDS(paste0(data_dir,
                   'SigPos.rds'))

NotSig <- readRDS(paste0(data_dir,
                   'NotSig.rds'))

facet_names <- 
  list('1'="1",'2'="",'3'="",'4'="",
       '5'="5",'6'="",'7'="",'8'="",'9'="",
       '10'="10",'11'="",'12'="",'13'="",'14'="",
       '15'="15",'16'="",'17'="",'18'="",'19'="",
       '20'="",'21'="",'22'="")

facet_labeller <- function(variable,value){
  return(facet_names[value])
}


#make manhattan plot
GWAS <- 
  ggplot(NotSig, aes(x = Pos, 
                     y = plog10, 
                     color = as.factor(Chr))) +
  geom_point(alpha = 0.3, 
             size = 0.5) +
  geom_point(data = SigPos, 
             aes(x = Pos, 
                 y = plog10),
             color = 'blue',
             shape = 3,
             size = 0.5) +
  facet_grid(~Chr, 
             scales = 'free_x', 
             space = 'free_x', 
             switch = 'x',
             labeller = facet_labeller) +
  scale_y_continuous(expand = c(0,0)) +
  scale_color_manual(values = rep(
    c("grey70", "grey30"), 
    22)) +
  labs(y = '-log10(p)', x = 'Chromosome') +
  guides(color = FALSE) + 
  theme_classic() +
  theme(plot.title = element_text(hjust = 0.5), 
        plot.subtitle = element_text(hjust = 0.5), 
        axis.text.x = element_blank(), 
        axis.ticks.x = element_blank(),
        axis.line.x = element_blank(), 
        strip.background = element_blank(),
        strip.text = element_text(size = 20),
        axis.text = element_text(size = 20),
        axis.title = element_text(size = 20,
                                  vjust = -5))
```


## Joint Frequency Spectrum
```{r,fig.height=5,fig.width=5}
#load file if it already exists
#see MakeFiles.Rmd for info on jointAF.txt
join <- readRDS(paste0(data_dir,
               'jointAF.rds'))

#load poisitons that reach -log10(p) > 6
sig.6 <- SigPos[,c('Chr','Pos')]
sig.6$sig <- 'Significant' #add dummy variable for joining
join <- merge(join, 
              sig.6, 
              by=c('Chr', 'Pos'), 
              all.x=T)
join <- join[which(join$JPT_MAF + join$NAG_MAF != 0),] #remove fixed
jj <- join[complete.cases(join),]
```

```{r}
SFS <- 
  ggplot(join, 
         aes(x = NAG_MAF, 
             y = JPT_MAF)) +
  geom_bin2d(binwidth = c(1/104, 1/104)) +
  geom_bin2d(data = jj,
             fill = 'blue',
             binwidth = c(1/104, 1/104), 
             alpha = 0.7) +
  scale_fill_gradient(name = 'log10\ncount',
                      low = "grey95",
                      high = "black",
                      trans = "log",
                      breaks = c(1,10,100,
                                 1000,10000,
                                 100000,1000000), 
                      labels = c('0','1', '2', 
                                 '3','4', 
                                 '5', '6')) +
  scale_x_continuous(expand = c(0.01,0.01), 
                     limits = c(-0.01,0.5)) +
  scale_y_continuous(expand = c(0.01,0.01), 
                     limits = c(-0.01,0.5)) + 
  labs(x = "Nagahama Frequency",
       y = "1kGP Frequency") +
  theme_classic() +
  theme(axis.line = element_blank(),
        axis.text = element_text(size = 20),
        axis.title.x = element_text(size = 20),
        axis.title.y = element_blank(),
        legend.text = element_text(size = 20),
        legend.title = element_text(size = 20),
        plot.margin = unit(c(0,0,1.5,0), "lines"))

freq_gw <- data.frame('freq' = join$JPT_MAF,
                      'SNPs' = 'All')
freq_sig <- data.frame('freq' = jj$JPT_MAF,
                       'SNPs' = 'Suspicious')
freq_hist <- rbind(freq_sig, freq_gw)

HIST <- 
  ggplot(data = freq_hist %>%
           filter(freq > 0),
         aes(x=freq, 
             fill = SNPs))+
  geom_histogram(binwidth = 1/104,
                 alpha=0.8, 
                 position="identity",
                 aes(y=..density..))+
  geom_hline(yintercept = 16.5, 
             color = 'white',
             size = 5) +
  coord_flip(ylim = c(0, 18)) +
  scale_fill_manual(values=c("blue", "grey30")) +
  scale_x_continuous(limits = c(-0.01,0.5),
                     expand = c(0.01, 0.01)) +
  scale_y_reverse(expand = c(0.01, 0.01), 
                  breaks = c(0,5,10,15,16.5,18),
                  labels = c('0','5','10','15','...','30'))+
  labs(y="Density") +
  theme_classic() +
  theme(legend.position = c(0.1, 0.8),
        axis.line = element_blank(),
        axis.ticks = element_blank(),
        axis.text.y = element_blank(),
        axis.text.x = element_text(size = 20),
        axis.title.x = element_blank(),
        plot.margin = unit(c(0,0,1.5,0), "lines"))


SFS_HIST <- 
  plot_grid(HIST, 
            SFS,
            nrow = 1, 
            rel_widths = c(3, 5), 
            align = 'h', 
            labels=c('','D'))

#rm(join)#save memory
#rm(SFS)
#B
```


## Mutation Spectrum enrichment
```{r,fig.height=3,fig.width=9}
#load mutation spectrum for suspicious SNPs
SNP <- readRDS(paste0(data_dir,
                   'SNP_Context.rds'))
#load mutation spectrum for control SNPs
rSNP <- readRDS(paste0(data_dir,
                   'rSNP_Context.rds'))

#make color scale comparable between plots
maxFreq <- max(SNP$Freq)

plotContext <- function(data, name){
  
plt <- ggplot(data, 
              aes(x = End, 
                  y = Start, 
                  fill = Freq)) +
    geom_tile() +
    facet_grid(.~Mut) +
    labs(y = name) +
    theme_classic() +
    theme(axis.title.x = element_blank(),
          axis.line = element_blank(),
          axis.text = element_text(size = 20),
          axis.title.y = element_text(size = 20),
          strip.placement = "outside",
          strip.background = element_blank(),
          strip.text = element_text(size = 20),
          legend.text = element_text(size = 20),
          legend.title = element_text(size = 20),
          panel.border = element_rect(colour = "black", 
                                      fill = NA, 
                                      size = 1),
          plot.margin = unit(c(0,0,0,1.5), "lines")) +
    scale_fill_gradient(limits = c(0, maxFreq),
                        low = 'white', 
                        high = 'blue',
                        guide = guide_legend(title = 'count',
                                             direction = 'vertical'))
return(plt)
}

SNPplt <- plotContext(SNP,'Suspicious')
rSNPplt <- plotContext(rSNP,'Control')

#extract legend
#https://github.com/hadley/ggplot2/wiki/Share-a-legend-between-two-ggplot2-graphs
g_legend <- function(a.gplot){
  tmp <- ggplot_gtable(ggplot_build(a.gplot))
  leg <- which(sapply(tmp$grobs, 
                      function(x) x$name) == "guide-box")
  legend <- tmp$grobs[[leg]]
  return(legend)}

mylegend <- g_legend(SNPplt)

Mut_Spect <- 
  plot_grid(
            plot_grid(
              SNPplt + 
                theme(legend.position = "none"),
              rSNPplt + 
                theme(legend.position = "none"),
              nrow = 2),
            mylegend,
            ncol = 2,
            rel_widths = c(1,0.1))

Mut_Spect


```
## Put all plots together and save
```{r,fig.height=11,fig.width=8}
#AB <- plot_grid(A, B,
#               rel_widths = c(1, 2.7))
fig1 <- plot_grid(GWAS,
                  Mut_Spect,
                  SFS,
               ncol = 1,
               rel_heights = c(1,2,4),
               labels = c('A','B','C'))

#figure takes time to save so skip if already done
#if(!file.exists(paste0(figure_dir,
#                       'Figure1.pdf'))){ 
ggsave(paste0(figure_dir,
              'Figure1.1.jpg'),
       fig1, 
       height = 14,
       width = 12)
#ggsave(paste0(figure_dir,
#              'Figure1.pdf'),
#       fig1,
#       height = 12,
#       width = 12)
#}
#fig1

```


```{r}
#save memory
rm(GWAS)
rm(fig1)
rm(A)
rm(B)
rm(AB)

```

#Figure 2
This plot illustrates the quality of the date from the 1000 Genomes Project over time.
Before we can plot this, we must download and parse the data.
See ProcessPlotData.Rmd for info

### Quality per population
```{r,fig.height=5,fig.width=5}
metaData <- readRDS(paste0(data_dir,
                           'metaData.rds'))

a <- ggplot(metaData, 
            aes(x = sample, 
                y = avg_qual,
                color = Population,
                shape = as.factor(Phase))) +
            geom_point() +
            geom_hline(yintercept = 30, 
                       linetype = 2, 
                       color = 'grey') +
            scale_colour_manual(values = MyColour, 
                                breaks = unique(metaData$Population)) +
            scale_x_discrete(expand = c(0.1,0)) +
            scale_shape_manual(values = c(1,3)) +
            guides(color = guide_legend(title = "Population",
                                        shape = FALSE,
                                        ncol = 1,
                                        override.aes = list(shape = 15,
                                                            size = 3))) +
            labs(title = "Data quality over time",
                 x = "Populations ranked by sequencing date",
                 y = "Average quality of mapped bases") +
            theme(plot.title = element_text(hjust = 0.5), 
                  axis.text.x = element_blank(), 
                  axis.ticks.x = element_blank())

a
```
### Quality by sequencing center
```{r,fig.height=5,fig.width=5}
b <- ggplot(metaData, 
            aes(x = SUBMISSION_DATE, 
                y = avg_qual, 
                color = CENTER_NAME, 
                shape = as.factor(Phase))) +
            geom_point() + 
            geom_hline(yintercept = 30, 
                       linetype = 2, 
                       color = 'grey') +
            scale_shape_manual(values = c(1,3),
                               name = 'Phase') +
            guides(color = guide_legend(title = "Sequencing\n    Center",
                                        override.aes = list(shape = 15,
                                                            size = 5))) +
            labs(y = "Average quality of mapped bases") +
            theme(axis.title.x = element_blank(),
                  axis.text.x = element_blank(),
                  axis.ticks.x = element_blank(), 
                  axis.line.x = element_blank())
b
```

### Sequencing technology use over time
```{r,fig.height=2,fig.width=6}
c <- ggplot(metaData, 
            aes(x = SUBMISSION_DATE, 
                y = INSTRUMENT_MODEL,
                shape = INSTRUMENT_MODEL)) +
            geom_point() +
            labs(x = "Individuals ranked by sequencing date", 
                 y = 'Sequencer') +
            guides(shape = guide_legend(title = "",
                                        label = FALSE,
                                        override.aes = list(alpha = 0))) +
            scale_x_date(date_labels = "%Y",
                         date_breaks = "1 year") +
            theme(axis.line.y = element_blank(),
                  axis.text.y = element_text(size = 10,
                                             hjust = 0))

c
```

Put it all together and save it
```{r,fig.height=11,fig.width=8}
d <- plot_grid(a,b,c, 
            nrow = 3, 
            labels = c('A', 'B', ''),
            rel_heights = c(4, 4, 1),
            align = 'v')
ggsave(paste0(figure_dir,
              "MapQualOverTime.jpg"),
       d,
       height = 14,
       width = 9)
ggsave(paste0(figure_dir,
              "MapQualOverTime.pdf"),
       d,
       height = 14,
       width = 9)

d
```

#Figure 3

This figure shows the overlap of sites associated to qualiy independantly in each single population regression.
See ProcessPlotData.Rmd for info on the overLap file

### Overlap of significant SNPs
```{r,fig.height=5,fig.width=5}
overLap <- readRDS(paste0(data_dir,
                           'overLap.rds'))

p1 <- ggplot(overLap, 
          aes(x = Pos,
              y = reorder(title, -Freq), 
              color = Population, 
              size = Pcat)) +
          geom_point(shape = 108) +
          scale_size(breaks = c(2, 3, 4), 
                     range = c(1, 3),
                     labels = c('6-8', '8-10', '10+'), 
                     guide = guide_legend(direction = "horizontal"),
                                          name = '-log10(p)') +
          scale_color_manual(breaks = overLap$Population,
                             values = MyColour) +
          guides(color = FALSE) +
          ylab('Population') +
          ggtitle('Overlap of Significant SNPs') +
          theme_classic() +
          theme(legend.position = c(0.85, 0.95),
                legend.box.background = element_rect(colour = "black"), 
                plot.title = element_text(hjust = 0.5), 
                axis.title.x = element_blank(),
                axis.text.x = element_blank(), 
                axis.ticks.x = element_blank(), 
                plot.margin = unit(c(1,1,0,1), "cm"))
p1

```

### Frequency of occurence
```{r,fig.height=2,fig.width=5}
#this is for the labels of the xaxis
xaxis <- overLap[!duplicated(overLap[,c('SigHits')]),]
xaxis <- xaxis[which(xaxis$SigHits < 11),]
xaxis <- rbind(xaxis,tail(overLap, 1))

p2 <- ggplot(overLap,
             aes(x = Pos,
                 y = SigHits, 
                 group=1))+
             geom_hline(yintercept = 1, 
                        color = 'grey', 
                        linetype = 2) +
             geom_hline(yintercept = 5,
                        color = 'grey',
                        linetype = 2) +
             geom_hline(yintercept = 10,
                        color = 'grey',
                        linetype = 2) +
             geom_line() +
             scale_y_continuous(limits = c(0,16),
                                breaks = c(1,5,10,15)) +
             scale_x_discrete(expand = c(0.01,0),
                              breaks  = xaxis$Pos, 
                              labels = xaxis$nrow) + 
             theme_classic() +
             theme(plot.title = element_text(hjust = 0.5),
                   plot.margin = unit(c(0,1,1,1), "cm"),
                   axis.text.x = element_text(angle = 60, 
                                              hjust = 1)) +
             labs(x = 'SNPs ranked by frequency of occurence and genomic position', 
                  y = 'Frequency\nof occurence')
p2
```

put it all together and save it
```{r,fig.height=11,fig.width=8}
p3 <- plot_grid(p1, p2, 
                ncol = 1, 
                align = "v", 
                rel_heights = c(3, 1))

ggsave(paste0(figure_dir,
              'SNPOverlap6.jpg'),
              p3, 
              height = 8, 
              width = 8)

ggsave(paste0(figure_dir,
              'SNPOverlap6.PDF'),
              p3, 
              height = 8, 
              width = 8)

p3
```


#Supplementary Figures

##Comparig models
```{r,fig.height=11,fig.width=8}
fitsName = 'fits_Significant_Positions_'
#load data
LFA <- fread(paste0(data_dir,fitsName,"LFA.txt"))
  
Pop <- fread(paste0(data_dir,fitsName,"Pop.txt"))
  
PCA <- fread(paste0(data_dir,fitsName,"PCA.txt"))

plt <- merge(LFA,Pop,by=c('rsID'),suffixes = c('LFA','PCA'))
plt <- merge(plt,PCA,by=c('rsID'))

p1a<-ggplot(plt, aes(b, bPCA)) +
  geom_abline(color='grey') +
  geom_point(shape=1, alpha=0.2) +
  labs(x='Pop', y='PCA')

p1b<-ggplot(plt, aes(b, bLFA)) +
  geom_abline(color='grey') +
  geom_point(shape=1, alpha=0.2) +
  labs(x='Pop', y='LFA') +
  ggtitle('Beta')

p1c<-ggplot(plt, 
            aes(bLFA, 
                bPCA)) +
    geom_abline(color='grey') +
  geom_point(shape=1, alpha=0.2) +
  labs(x='LFA', y='PCA')

p2a<-ggplot(plt, 
            aes(devdiff, 
                devdiffPCA))+
  geom_abline(color='grey')+
  geom_point(shape=1, alpha=0.2)+
  labs(x='Pop', y='PCA')
p2b<-ggplot(plt, 
            aes(devdiff, 
                devdiffLFA))+
  geom_abline(color='grey')+
  geom_point(shape=1, alpha=0.2)+
  labs(x='Pop', y='LFA')+
  ggtitle('Deviance')
p2c<-ggplot(plt, 
            aes(devdiffLFA, 
                devdiffPCA))+
  geom_abline(color='grey')+
  geom_point(shape=1, alpha=0.2)+
  labs(x='LFA', y='PCA')

plot_grid(p1b,p2b,
          p1a,p2a,
          p1c,p2c,
          nrow=3)

ggsave(paste0(figure_dir,fitsName,
              'CompareModels.jpg'),
              height=10, 
              width=8)
```


## Comparing 1kGP to Resequenced Han
###Single Population Test
```{r,fig.height=5,fig.width=5}
#using the sites associated to Quality in the single population test
HanKgpAF <- readRDS(paste0(data_dir,
                   'HanKgpAF.rds'))

ggplot(HanKgpAF, 
       aes(x = hanAF/2, 
           y = kgpAF/2))+
  geom_bin2d(bins = 83)+
  scale_fill_distiller(palette = "Spectral")+
  labs(x = 'Resequenced Han', 
       y = '1000 Genomes Project')+
  ggtitle(paste0('Joint frequency plot of ',nrow(HanKgpAF),' suspicious SNPs\nfrom single population test'))

ggsave(paste0(figure_dir,'Han_1kGP_SFS_singlePop.jpg'),height=10, width=10)

```
##Full Model
```{r,fig.height=5,fig.width=5}
#Load joint frequency for SNPs
##from plink --freq
kgpAF <- fread(paste0(data_dir,'Significant_Positions.frq'))
hanAF <- fread(paste0(data_dir,'Significant_Positions_Han.frq'))
combo <- merge(hanAF, kgpAF, 
               by=c('CHR','SNP'), 
               suffixes = c('.HAN','.KGP'), 
               all = T)
combo[is.na(combo)] <- 0
#keep only sites present in 83 KGP samples
combo <- combo[which(combo$MAF.KGP > 0),]
combo$diff <- abs(combo$MAF.KGP - combo$MAF.HAN)
nHan <- nrow(combo[which(combo$MAF.HAN > 0),])

ggplot(combo, 
        aes(x = MAF.HAN, 
            y = MAF.KGP)) +
     geom_bin2d(bins = 100) +
     scale_fill_distiller(palette = "Spectral")+
    labs(x = 'Resequenced Han', 
         y = '1000 Genomes Project')+
    ggtitle(paste0('Joint frequency spectrum of \n',nrow(combo),' suspicious variants from combined test\n(',nHan, ' variants present in Han)'))

ggsave(paste0(figure_dir,'Han_1kGP_SFS_FullModel.jpg'),height=10, width=10)


```

##NYGC Frequencies
```{bash}
for f in `seq 1 22`; \
do vcftools \
--gzvcf ./CCDG_13607_B01_GRM_WGS_2019-02-19_chr${f}.recalibrated_variants.vcf.gz \
--freq --out ./chr${f}; \
done
```

##NYGC ReSequence Data
```{r,fig.height=5,fig.width=5}
#Load joint frequency for SNPs
##from plink --freq
kgpAF <- fread('/Users/luke/genomes/genomes/hg19/Freq/genomeWide_1kGP.frq',
               col.names = c('Chr','Pos','N_Alleles','N_Counts','AF1','AF2'))
kgpAF <- kgpAF %>%
  separate(AF1, c("Allele1", "AF1"), ":")
kgpAF <- kgpAF %>%
  separate(AF2, c("Allele2", "AF2"), ":")
nygcAF <- fread(paste0(data_dir,'NYGC/genomeWide_NYGC_lifted_1-9.frq'),
               col.names = c('Chr','Pos','PosPlus1','N_Counts','AF1','Af2'))
nygcAF <- nygcAF %>%
  separate(AF1, c("Allele1", "AF1"), ":")
nygcAF$Chr <- as.numeric(as.character(gsub(pattern = 'chr',replacement = '',x = nygcAF$Chr)))

sigPos <- fread(paste0(data_dir,'Significant_Positions.txt'), 
                col.names = c('Chr','Pos'))

strictMask <- fread(paste0(data_dir,
                           '1kGP_StrictMask.bed'),
            col.names = c('Chr','Pos','Pos.1','rsID'))

combo <- merge(nygcAF, kgpAF, 
               by=c('Chr','Pos'),
               suffixes = c('.NYGC','.KGP'), 
               all = T)

combo <- combo[which(combo$Allele1.KGP == combo$Allele1.NYGC),]

combo$AF1.NYGC <- as.numeric(as.character(combo$AF1.NYGC))
combo$AF1.KGP <- as.numeric(as.character(combo$AF1.KGP))

combo[is.na(combo)] <- 0
combo$diff <- abs(combo$AF1.KGP - combo$AF1.NYGC)


sig <- merge(combo, sigPos, by=c('Chr','Pos'))

combo <- anti_join(combo,
                   strictMask,
                   by = c('Chr','Pos'))

combo <- combo[which((combo$AF1.NYGC > 0) &
                       (combo$AF1.KGP > 0)),]
combo <- combo[which((combo$AF1.NYGC < 1) &
                       (combo$AF1.KGP < 1)),]

sfs <- ggplot(sig, 
        aes(x = 1-AF1.NYGC, 
            y = 1-AF1.KGP)) +
     geom_bin2d(bins=250) +
     scale_fill_distiller(palette = "Spectral",
                          trans = "log",
                          breaks = c(5,50,500,5000))+
     scale_y_continuous(limits = c(0,1),
                        expand = c(0.01, 0.01)) +
    labs(x = 'Resequenced NYGC', 
         y = '1000 Genomes Project') +
    ggtitle('Suspicious Variants in Chromosome 1-9')

sfs

ggsave(paste0(figure_dir,'NYGC_Chr1-9.jpg'),height=10, width=10)

allsfs <- ggplot(combo, 
        aes(x = 1-AF1.NYGC, 
            y = 1-AF1.KGP)) +
     geom_bin2d(bins=250) +
     scale_fill_distiller(palette = "Spectral",
                          trans = "log",
                          breaks = c(1,10,100,1000,10000,100000,1000000))+
     scale_y_continuous(limits = c(0,1),
                        expand = c(0.01, 0.01)) +
    labs(x = 'Resequenced NYGC', 
         y = '1000 Genomes Project') +
    ggtitle('All Variants in Chromosome 1-9')



allsfs

ggsave(paste0(figure_dir,'NYGC_Chr1-9_all.jpg'),height=10, width=10)

sfsPlus <- ggplot(combo, 
        aes(x = 1-AF1.NYGC, 
            y = 1-AF1.KGP)) +
     geom_bin2d(bins=250) +
     geom_point(data = sig, size = 0.8, alpha = 0.7, shape = 3) +
     scale_fill_distiller(palette = "Spectral",
                          trans = "log",
                          breaks = c(1,10,100,1000,10000,100000,1000000))+
     scale_y_continuous(limits = c(0,1),
                        expand = c(0.01, 0.01)) +
    labs(x = 'Resequenced NYGC', 
         y = '1000 Genomes Project') +
    ggtitle('All Variants in Chromosome 1-9')



sfsPlus

ggsave(paste0(figure_dir,'NYGC_Chr1-9_allPlus.jpg'),height=10, width=10)
```
```{r}
hist <- ggplot(sig, aes(x=diff)) +
            geom_histogram(binwidth = 0.01) +
            scale_x_continuous(limits = c(0,1),
                             expand = c(0.01, 0.05)) +
            scale_y_reverse(limits=c(3000,0)) +
            labs(x = "Difference in Frequency",
                 y = "") +
            coord_flip() +
            theme_classic() +
            theme(axis.line = element_blank())

```

```{r,fig.height=11,fig.width=8}
AF30 <- readRDS(paste0(data_dir,
                   'AF30.rds'))
AF30$rate <- AF30$oAF/AF30$uAF

af1 <- ggplot(AF30, aes(x=uAF, y=oAF))+
  geom_bin2d(bins=200)+
  scale_fill_distiller(palette = "Spectral")+
  geom_abline(intercept=0, slope = 2279/225, linetype=2, color='black')+
  geom_abline(intercept=0, slope = 2279/450, linetype=3, color='red')+
  labs(x='Samples with Q < 30',y='Samples with Q > 30')

af2 <- ggplot(AF30[which(AF30$oAF <100),], aes(x=uAF, y=oAF))+
  geom_bin2d(bins=100)+
  scale_fill_distiller(palette = "Spectral")+
  geom_abline(intercept=0, slope = 2279/225, linetype=2, color='black')+
  geom_abline(intercept=0, slope = 2279/450, linetype=3, color='red')+
  labs(x='Samples with Q < 30',y='Samples with Q > 30')

plot_grid(af1, af2,ncol=1)

ggsave(paste0(figure_dir,'OverUnder30.jpg'),height=11,width=8)
ggsave(paste0(figure_dir,'OverUnder30.pdf'),height=11,width=8)

print(nrow(AF30[which(AF30$rate <= 2279/450),])/nrow(AF30))
print(nrow(AF30[which(AF30$rate > 2279/450),])/nrow(AF30))
```
##Over Under 30 for published variants
```{r}
pub<-fread(paste0(data_dir, 'fits_PublishedVariants_LFA.txt'))
pub <- merge(pub, AF30, by='rsID')
pub$text <- paste(pub$rsID, 'p ',round(pub$dev_logp, 2))
library(ggrepel)
ggplot(pub, aes(x=uAF, y=oAF, label=text))+
     geom_point()+
     geom_text_repel(size=3)+
     geom_abline(intercept=0, slope = 2279/225, linetype=2, color='black')+
     geom_abline(intercept=0, slope = 2279/450, linetype=3, color='red')+
     labs(x='Samples with Q < 30',y='Samples with Q > 30')+
     ggtitle('Published Variants')
ggsave(paste0(figure_dir,'Published_OverUnder30.jpg'),height=7,width=8)
pub$ratio <- pub$oAF/pub$uAF
pub$limit <- pub$ratio > 2279/450
write.table(pub,paste0(figure_dir,'Published_OverUnder30.txt'),)

```

```{r,fig.height=5,fig.width=10}
#load file
omni <- readRDS(paste0(data_dir,'omni.rds'))

ggplot(omni, 
       aes(x = (2504 - num_zeros)/2504, 
           y = dev_logpLFA)) +
  geom_point(shape = 1) +
  labs(x = 'Global Allele Frequency',
       y = '-log10(p)') +
  ggtitle(paste0(nrow(omni),' SNPs present in the Omni 2.5 Microarray'))

ggsave(paste0(figure_dir,'Omni_AF.jpg'), 
       height = 5, 
       width = 10)

```


```{r}
sessionInfo()
```

