---
title: "Legacy Data Confounds Modern Genomics Studies"
author: 
- name: Luke Anderson-Trocm√©, 
  affiliation:
  - &cruk Department of Human Genetics, McGill University,  Montreal, Canada
date: '`r format(Sys.Date(), "%Y-%B-%d")`'
output:
  html_notebook:
    df_print: paged
    code_folding: show
    toc: yes
    toc_float: 
      collapsed: false
      smooth_scroll: false
---
  

load libraries
```{r, message = FALSE}
library(ggplot2)
library(RColorBrewer)
library(plotly)
library(colorspace)
library(cowplot)
library(data.table)
library(tidyverse)
library(purrr)

#Directories
filterPath <- '/Users/luke/genomes/genomes/1kGP_4bedFiltered/'
miscPath <- '~/Documents/QualityPaper/Misc/'
figurePath <- '~/Documents/QualityPaper/Figures/'
```

# Figure 1
## Reverse GWAS in JPT
```{r}
#Load single population regression results
Reg <- fread(paste0(miscPath,
                    '1KGP.GenomeWide.Regression_JPT.csv'), 
             col.names = c('Chr','Pos',
                           'Population','deviance'), 
             colClasses = c('numeric', 'numeric', 
                            'character', 'numeric'))

#Compute P values from deviance from null model
Reg$plog10 <- -log10(pchisq(Reg$deviance, 
                            df = 1, 
                            lower.tail = FALSE))
Reg$p<- pchisq(Reg$dev,
               df = 1, 
               lower.tail = FALSE)

#split data into sig / not sig for plotting
SigPos <- Reg[which(Reg$plog10 > 6),]
NotSig <- Reg[which(Reg$plog10 < 6),]
rm(Reg) #save memory

#make manhattan plot
GWAS <- ggplot(NotSig, aes(x = Pos, 
                           y = plog10, 
                           color = as.factor(Chr))) +
              geom_point(alpha = 0.3, 
                         size = 1) +
              geom_point(data = SigPos, 
                         aes(x = Pos, 
                             y = plog10,),
                         color = 'black',
                         shape = 3) +
              facet_grid(~Chr, 
                           scales = 'free_x', 
                           space = 'free_x', 
                           switch = 'x') +
              scale_y_continuous(expand = c(0,0)) +
              scale_fill_brewer(palette = 'Spectral') +
              labs(y = '-log10(p)', x = 'Chromosome') +
              guides(color = FALSE) + 
              theme_classic() +
              theme(plot.title = element_text(hjust = 0.5), 
                    plot.subtitle = element_text(hjust = 0.5), 
                    axis.text.x = element_blank(), 
                    axis.ticks.x = element_blank(),
                    axis.line.x = element_blank(), 
                    strip.background = element_blank(),
                    strip.text.x = element_text(size = 6))
GWAS
```


## Joint Frequency Spectrum
```{r}
  
if(!file.exists(paste0(miscPath,
                       'jointAF.txt'))) {
  #Load Allele Frequency files
  AF.1kGP <- paste0(filterPath,
                    '1kGP_GenomeWide_JPT_1.frq')
  JPT <- fread(
    AF.1kGP,
    fill = TRUE,
    col.names = c('Chr', 'Pos', 'N_ALLELES',
                  'N_CHR', 'JPT_AF', 'JPT_MAF')
  )
  AF.NAG <- paste0(filterPath,
                   'NAG_GenomeWide.4bed_filtered.freq.frq')
  NAG <- fread(
    AF.NAG,
    fill = TRUE,
    col.names = c('Chr', 'Pos', 'N_ALLELES',
                  'N_CHR', 'NAG_AF', 'NAG_MAF')
  )
  
  #merge them together to get joint frequency spectrum
  join <- merge(JPT[which(JPT$JPT_AF <= 1), ],
                NAG[which(NAG$NAG_AF <= 1), ],
                by = c('Chr', 'Pos'), all = TRUE)
  join[is.na(join)] <- 0 #missing positions are frequency of 0
  
  #REMOVE  HWE SITES
  hwe_1kGP <- paste0(filterPath,
                     '1kGP_GenomeWide.4bed_filtered_sig6.freq.hwe')
  JPT_hwe <- fread(hwe_1kGP)
  hwe_NAG <- paste0(filterPath,
                    'NAG_GenomeWide.4bed_filtered_sig6.freq.hwe')
  NAG_hwe <- fread(hwe_NAG)
  
  hwe <- merge(NAG_hwe[, c('V1', 'V2')],
               JPT_hwe[, c('V1', 'V2')],
               by = c('V1', 'V2'),
               all = TRUE)
  #remove sites from AF that don't meet HWE
  join <- join[!paste(join$CHROM, join$POS)
               %in% c(paste(hwe$V1, hwe$V2)),]
  
  join <- join[, c('Chr', 'Pos', 'JPT_MAF',
                   'JPT_AF', 'NAG_MAF', 'NAG_AF')]
  #make numeric
  join <- sapply(join, as.numeric)
  join <- join[complete.cases(join), ]
  
  rm(NAG) #free up memory
  rm(JPT)
  rm(hwe)
  #save to file
  write.table(join,
              paste0(miscPath,
                     'jointAF.txt'))
}
#to save time, load file if it already exists
if(file.exists(paste0(miscPath,
                      'jointAF.txt'))){
join <- fread(paste0(miscPath,
                     'jointAF.txt'))}

#load poisitons that reach -log10(p) > 6
sig.6 <- SigPos[,c('Chr','Pos')]
sig.6$sig <- 'Significant' #add dummy variable for joining
join <- merge(join, 
              sig.6, 
              by=c('Chr', 'Pos'), 
              all.x=T)
join <- join[which(join$JPT_MAF + join$NAG_MAF != 0),] #remove fixed
jj <- join[complete.cases(join),]

SFS <- ggplot(join, aes(x = NAG_MAF, y = JPT_MAF)) +
          geom_bin2d(binwidth = c(1/884, 1/104)) +
          geom_point(data = jj, size = 0.8, alpha = 0.7, shape = 3) +
          scale_fill_distiller(palette = "Spectral",
                               trans = "log",
                               breaks = c(1,10,100,1000,
                                          10000,100000,1000000), 
                               labels = c('1','10', '100', '1,000', 
                                          '10,000', '100,000', '1,000,000')) +
          scale_x_continuous(expand = c(0.01,0.01)) +
          scale_y_continuous(expand = c(0.01,0.01)) + 
          xlab(label = "Nagahama") +
          theme_classic() +
          theme(axis.line = element_blank(), 
                axis.title.y = element_blank(), 
                axis.text.y = element_blank(), 
                axis.ticks.y = element_blank())

HIST <- ggplot(jj, aes(x = JPT_MAF)) +
            geom_histogram(binwidth = 0.01, 
                         fill = 'grey') +
            scale_x_continuous(limits = c(0,1),
                             expand = c(0.01, 0.01)) +
            scale_y_reverse(expand = c(0.01, 0.01), 
                          breaks = c(10, 50, 100)) +
            labs(x = "1000 Genomes Project",
                 y = "") +
            coord_flip() +
            theme_classic() +
            theme(axis.line = element_blank(),
                  axis.text.x = element_text(angle = 90))

B <- plot_grid(HIST, 
               SFS,
               nrow = 1, 
               rel_widths = c(1, 5), 
               align = 'h', 
               labels=c('','B'))

rm(join)#save memory
rm(SFS)
B
```


## Mutation Spectrum enrichment
```{r}
#get positions of significant SNPs
sig.pos <- SigPos[,c('Chr','Pos')]
write.table(sig.pos, 
            file=paste0(miscPath,
                        'sig.pos.txt'), 
            col.names = FALSE, 
            row.names = FALSE, 
            quote = FALSE, 
            sep = '\t')

#get positions of random SNPs as control
set.seed(123)
random.pos <- NotSig[sample(nrow(NotSig), 
                            nrow(sig.6)),
                     c('Chr','Pos')]
write.table(random.pos, 
            file = paste0(miscPath,
                        'random.pos.txt'), 
            col.names = FALSE, 
            row.names = FALSE, 
            quote = FALSE, 
            sep = '\t')
rm(NotSig) #save memory
```
extract context for snps of interest
```{bash}
#Using GenomeWide.AncestralContext.txt
#To get mutation context, use $path/AncestralContext.py
#compare first two columns in both files, extract matches
path='/Users/luke/Documents/QualityPaper/Misc/'
if [ ! -f $path/random.pos.context.txt ]; then
  echo "random pos context"
  awk 'NR==FNR{a[$1,$2];next} ($1,$2) in a' \
  $path/random.pos.txt \
  $path/GenomeWide.AncestralContext.txt \
  > $path/random.pos.context.txt
fi  
  
if [ ! -f $path/sig.pos.context.txt ]; then
  echo "sig pos context"
  awk 'NR==FNR{a[$1,$2];next} ($1,$2) in a' \
  $path/sig.pos.txt \
  $path/GenomeWide.AncestralContext.txt \
  > $path/sig.pos.context.txt
fi 
```
Load mutation context and plot enrichment
```{r}
contextHeader = c('Chr','Pos','Ref','Alt','Flip','Context')

#load SNP context
SNP <- read.table(paste0(miscPath,
                         'sig.pos.context.txt'), 
                sep = '\t', 
                header  = FALSE,
                col.names = contextHeader)
rSNP <- read.table(paste0(miscPath,
                          'random.pos.context.txt'), 
                sep = '\t',
                header = FALSE,
                col.names = contextHeader)

ReverseComp <- function(x)
        chartr("ATGC","TACG",
        sapply(
          lapply(
          strsplit(x, NULL), 
          rev), 
          paste, 
          collapse = ""))

processContext <- function(data){
  
#get reverse complement of context, ref and alt
data$Rev<-ReverseComp(as.character(data$Context))
data$RevRef<-ReverseComp(as.character(data$Ref))
data$RevAlt<-ReverseComp(as.character(data$Alt))

#only get one half (to fold over)
AC<-data[which(data$Ref 
               %in% 
                 c('A','C')),]
TG<-data[which(data$Ref 
               %in% 
                 c('T','G')),]
#fold over
TG$Context <- TG$Rev
TG$Ref <- TG$RevRef
TG$Alt <- TG$RevAlt
#bind them together
data <- rbind(AC,TG)
#combine it all together to get a context
data$Mut <- paste0(data$Ref,'->', 
                substr(data$Context,1,1),
                data$Alt,
                substr(data$Context,3,3))
#get the counts of each bin
binCount <- as.data.frame(table(data$Mut))
#process for plotting
binCount$Start <- substr(binCount$Var1,4,4)
binCount$End <- substr(binCount$Var1,6,6)
binCount$Mut <- paste(substr(binCount$Var1,1,3),
                      substr(binCount$Var1,5,5))
return(binCount)
}

SNP <- processContext(SNP)
rSNP <- processContext(rSNP)
maxFreq <- max(SNP$Freq) #make color scale comparable between plots

plotContext <- function(data, name){
  
plt <- ggplot(data, 
              aes(x = End, 
                  y = Start, 
                  fill = Freq)) +
    geom_tile() +
    facet_grid(Mut~., 
               switch = "y") +
    labs(subtitle = name) +
    theme_classic() +
    theme(axis.title = element_blank(),
          axis.line = element_blank(),
          legend.position = "bottom",
          strip.placement = "outside",
          strip.background = element_blank(),
          panel.border = element_rect(colour = "black", 
                                      fill = NA, 
                                      size = 1)) +
    scale_fill_gradient(limits = c(0, maxFreq),
                        low = 'white', 
                        high = 'blue',
                        guide = guide_legend(title = 'Count',
                                             title.position = "top"))
return(plt)
}

SNPplt <- plotContext(SNP,'Suspicious SNPs')
rSNPplt <- plotContext(rSNP,'Control SNPs')

#extract legend
#https://github.com/hadley/ggplot2/wiki/Share-a-legend-between-two-ggplot2-graphs
g_legend <- function(a.gplot){
  tmp <- ggplot_gtable(ggplot_build(a.gplot))
  leg <- which(sapply(tmp$grobs, 
                      function(x) x$name) == "guide-box")
  legend <- tmp$grobs[[leg]]
  return(legend)}

mylegend <- g_legend(SNPplt)

A <- plot_grid(plot_grid(
              SNPplt + 
                theme(legend.position = "none"),
              rSNPplt + 
                theme(legend.position = "none")),
              mylegend,
              nrow = 2,
              rel_heights = c(1, 0.1),
              labels = c('A'))

A
```
## Put all plots together and save
```{r}
AB <- plot_grid(A, B,
               rel_widths = c(1, 2.7))
fig1 <- plot_grid(AB,
               GWAS,
               ncol = 1,
               rel_heights = c(2, 1),
               labels = c('','C'))

#figure takes time to save so skip if already done
if(!file.exists(paste0(figurePath,
                       'Figure1.pdf'))){ 
ggsave(paste0(figurePath,
              'Figure1.jpg'),
       fig1, 
       height = 12,
       width = 12)
ggsave(paste0(figurePath,
              'Figure1.pdf'),
       fig1,
       height = 12,
       width = 12)
}
#save memory
rm(GWAS)
rm(fig1)
rm(A)
rm(B)
rm(AB)
```

#Figure 2

This plot illustrates the quality of the date from the 1000 Genomes Project over time.
Before we can plot this, we must download and parse the data.

```{r}
BasPath = '/Users/luke/genomes/bas_file/1000GenomesBAS/'

#load bas_dt
#see 1_make_quality_dt.Rmd for making this file
load('/Users/luke/Documents/QualityPaper/rick/bas_dt')

#get the index files for extra meta data
#wget http://ftp.1000genomes.ebi.ac.uk/vol1/ftp/phase3/20130502.phase3.analysis.sequence.index
index <- read.table(paste0(BasPath, 
                           '20130502.phase3.analysis.sequence.index'),
               comment.char = "?", 
               fill = TRUE,
               sep = "\t", 
               header = TRUE,
               stringsAsFactors = FALSE)
#keep columns of interest
index <- unique(index[,c('SAMPLE_NAME',
                         'CENTER_NAME',
                         'SUBMISSION_DATE',
                         'INSTRUMENT_MODEL')])

#Phase 1 samples, list downloaded from http://www.internationalgenome.org/data-portal/sample
phase <- read.table(paste0(miscPath,
                           'Phase1_IDs.txt'))
phase$Phase = 1

#merge tables
bas_dt <- merge(bas_dt, 
                index, 
                by.x = "sample", 
                by.y = "SAMPLE_NAME",
                all.x = TRUE)
bas_dt <- merge(bas_dt, 
                phase, 
                by.x = "sample", 
                by.y = "V1", 
                all.x = TRUE)
bas_dt[which(is.na(bas_dt$Phase)),]$Phase <- 3
#clean up the text
bas_dt$SUBMISSION_DATE <- gsub(" 00:00:00",
                               "",
                               bas_dt$SUBMISSION_DATE)
bas_dt$SUBMISSION_DATE <- as.Date(bas_dt$SUBMISSION_DATE, 
                                  "%Y-%m-%d")
bas_dt$CENTER_NAME<-gsub('Illumina',
                         'ILLUMINA',
                         bas_dt$CENTER_NAME)
bas_dt$INSTRUMENT_MODEL <-gsub('Illumina',
                               '',
                               bas_dt$INSTRUMENT_MODEL)
bas_dt$INSTRUMENT_MODEL <-gsub('Genome Analyzer',
                               'GA',
                               bas_dt$INSTRUMENT_MODEL)
bas_dt$INSTRUMENT_MODEL <-gsub('HiSeq 2000',
                               'HS 2k',
                               bas_dt$INSTRUMENT_MODEL)
#remove missing data
bas_dt<-bas_dt[complete.cases(bas_dt),]

#Get the average sequencing date per population
#Used for ranking the populations
Day<-aggregate(bas_dt$SUBMISSION_DATE, 
               list(bas_dt$Population), 
               mean)
#add it to data
metaData<-merge(bas_dt, 
                Day, 
                by.x = 'Population', 
                by.y = 'Group.1')
#reorder data by average date then quality
metaData <- metaData[with(metaData, 
                          order(x,
                                Population, 
                                avg_qual)),]
#set factor with order in mind (for plotting)
metaData$sample <- factor(metaData$sample, 
                          levels = unique(metaData$sample))

#Load the custom population colors
#Hex colors taken from https://www.nature.com/articles/nature15393
colors<-read.table(paste0(miscPath,
                          'NamePopBigPop_color.html.txt'), 
                   comment.char = '>',
                   col.names = c('Population',
                                 'hex',
                                 'BigPop'))
MyColour <- as.character(colors$hex)
names(MyColour) <- colors$Population

metaData
```

Make plots

### Quality per population
```{r}

a <- ggplot(metaData, 
            aes(x = sample, 
                y = avg_qual,
                color = Population,
                shape = as.factor(Phase))) +
            geom_point() +
            geom_hline(yintercept = 30, 
                       linetype = 2, 
                       color = 'grey') +
            scale_colour_manual(values = MyColour, 
                                breaks = unique(metaData$Population)) +
            scale_x_discrete(expand = c(0.1,0)) +
            scale_shape_manual(values = c(1,3)) +
            guides(color = guide_legend(title = "Population",
                                        shape = F,
                                        ncol = 1,
                                        override.aes = list(shape = 15,
                                                            size = 3))) +
            labs(title = "Data quality over time",
                 x = "Populations ranked by sequencing date",
                 y = "Average quality of mapped bases") +
            theme(plot.title = element_text(hjust = 0.5), 
                  axis.text.x = element_blank(), 
                  axis.ticks.x = element_blank())

a
```
### Quality by sequencing center
```{r}
b <- ggplot(metaData, 
            aes(x = SUBMISSION_DATE, 
                y = avg_qual, 
                color = CENTER_NAME, 
                shape = as.factor(Phase))) +
            geom_point() + 
            geom_hline(yintercept = 30, 
                       linetype = 2, 
                       color = 'grey') +
            scale_shape_manual(values = c(1,3),
                               name = 'Phase') +
            guides(color = guide_legend(title = "Sequencing\n    Center",
                                        override.aes = list(shape = 15,
                                                            size = 5))) +
            labs(y = "Average quality of mapped bases") +
            theme(axis.title.x = element_blank(),
                  axis.text.x = element_blank(),
                  axis.ticks.x = element_blank(), 
                  axis.line.x = element_blank())
b
```

### Sequencing technology use over time
```{r}
c <- ggplot(metaData, 
            aes(x = SUBMISSION_DATE, 
                y = INSTRUMENT_MODEL,
                shape = INSTRUMENT_MODEL)) +
            geom_point() +
            labs(x = "Individuals ranked by sequencing date", 
                 y = 'Sequencer') +
            guides(shape = guide_legend(title = "",
                                        label = FALSE,
                                        override.aes = list(alpha = 0))) +
            scale_x_date(date_labels = "%Y",
                         date_breaks = "1 year") +
            theme(axis.line.y = element_blank(),
                  axis.text.y = element_text(size = 10,
                                             hjust = 0))

c
```

Put it all together and save it
```{r}
d <- plot_grid(a,b,c, 
            nrow = 3, 
            labels = c('A', 'B', ''),
            rel_heights = c(4, 4, 1),
            align = 'v')
ggsave(paste0(figurePath,
              "MapQualOverTime.jpg"),
       d,
       height = 14,
       width = 9)
ggsave(paste0(figurePath,
              "MapQualOverTime.pdf"),
       d,
       height = 14,
       width = 9)
```

#Figure 3

This figure shows the overlap of sites associated to qualiy independantly in each single population regression.
```{r}
phase1 <- c('YRI','LWK','CHB','CHD','JPT','CEU','LWK')
#read single population regressions
Reg <- fread(paste0(miscPath,
                    'AllRegressionsOver10.txt'), 
             col.names = c('Chr','Pos',
                           'Population','deviance'), 
             colClasses = c('numeric', 'numeric', 
                            'character', 'numeric'))
#Get P-Values and significant sites
Reg$plog10 <- -log10(pchisq(Reg$deviance, 1, 
                            lower.tail = FALSE))
Reg<-Reg[which(Reg$plog10 >= 6),]
#Count number of populations where a variant is significant
NumPop <- as.data.frame(table(Reg$Chr,Reg$Pos))
NumPop <- NumPop[which(NumPop$Freq>1),]
names(NumPop) <- c('Chr','Pos', 'SigHits')
NumPop$Chr<-as.numeric(as.character(NumPop$Chr)) 
NumPop$Pos<-as.numeric(as.character(NumPop$Pos))

m <- merge(Reg,
           NumPop, 
           by = c('Chr','Pos'))
#count number of significant sites per population
n <- as.data.frame(table(m$Pop))
#put it all together for plotting
overLap <- unique(merge(m,n,
                        by.x = 'Population',
                        by.y = 'Var1'))
#include number of significant sites in titles
overLap$title = paste0(overLap$Pop,
                       ' : ', 
                       overLap$Freq)
#add asterix for Phase 1 pops
Phase1Title <- paste('*',
                     overLap[which(overLap$Pop 
                                   %in% phase1),]$title)
overLap[which(overLap$Pop
              %in% phase1),]$title <- Phase1Title
#set order by SigHits, Chromosome and Position
overLap <- overLap[with(overLap, 
                        order(-SigHits,
                              Chr,
                              Pos)),]
overLap$Pos <- factor(overLap$Pos, 
                      levels = unique(overLap$Pos))
#this is for keeping track of the number of variants
overLap$nrow <- seq(length = nrow(overLap))
#this is for discrete point size
overLap$Pcat <- 4
overLap[which(overLap$plog10 < 8),]$Pcat <- 2
overLap[which((overLap$plog10 > 8) & 
              (overLap$plog10 < 10)),]$Pcat <- 3
#this is for the labels of the xaxis
xaxis <- overLap[!duplicated(overLap[,c('SigHits')]),]
xaxis <- xaxis[which(xaxis$SigHits < 11),]
xaxis <- rbind(xaxis,tail(overLap, 1))

overLap
```

### Overlap of significant SNPs
```{r}
p1 <- ggplot(overLap, 
          aes(x = Pos,
              y = reorder(title, -Freq), 
              color = Population, 
              size = Pcat)) +
          geom_point(shape = 108) +
          scale_size(breaks = c(2, 3, 4), 
                     range = c(1, 3),
                     labels = c('6-8', '8-10', '10+'), 
                     guide = guide_legend(direction = "horizontal"),
                                          name = '-log10(p)') +
          scale_color_manual(breaks = overLap$Population,
                             values = MyColour) +
          guides(color = FALSE) +
          ylab('Population') +
          ggtitle('Overlap of Significant SNPs') +
          theme_classic() +
          theme(legend.position = c(0.85, 0.95),
                legend.box.background = element_rect(colour = "black"), 
                plot.title = element_text(hjust = 0.5), 
                axis.title.x = element_blank(),
                axis.text.x = element_blank(), 
                axis.ticks.x = element_blank(), 
                plot.margin = unit(c(1,1,0,1), "cm"))
p1

```

### Frequency of occurence
```{r}
p2 <- ggplot(overLap,
             aes(x = Pos,
                 y = SigHits, 
                 group=1))+
             geom_hline(yintercept = 1, 
                        color = 'grey', 
                        linetype = 2) +
             geom_hline(yintercept = 5,
                        color = 'grey',
                        linetype = 2) +
             geom_hline(yintercept = 10,
                        color = 'grey',
                        linetype = 2) +
             geom_line() +
             scale_y_continuous(limits = c(0,16),
                                breaks = c(1,5,10,15)) +
             scale_x_discrete(expand = c(0.01,0),
                              breaks  = xaxis$Pos, 
                              labels = xaxis$nrow) + 
             theme_classic() +
             theme(plot.title = element_text(hjust = 0.5),
                   plot.margin = unit(c(0,1,1,1), "cm"),
                   axis.text.x = element_text(angle = 60, 
                                              hjust = 1)) +
             labs(x = 'SNPs ranked by frequency of occurence and genomic position', 
                  y = 'Frequency\nof occurence')
p2
```

put it all together and save it
```{r}
p3 <- plot_grid(p1, p2, 
                ncol = 1, 
                align = "v", 
                rel_heights = c(3, 1))

ggsave(paste0(figurePath,
              'SNPOverlap6.jpg'),
              p3, 
              height = 10, 
              width = 10)

ggsave(paste0(figurePath,
              'SNPOverlap6.PDF'),
              p3, 
              height = 10, 
              width = 10)

```

```{r}
sessionInfo()
```

