---
title: "Process Regression Results"
author: 
- name: Luke Anderson-Trocmé, 
  affiliation:
  - &cruk Department of Human Genetics, McGill University,  Montreal, Canada
date: '`r format(Sys.Date(), "%Y-%B-%d")`'
output:
  html_notebook:
    df_print: paged
    code_folding: show
    toc: yes
    toc_float: 
      collapsed: false
      smooth_scroll: false
---
  
##Load Libraries
```{r}
library(dplyr)
library(magrittr)
library(data.table)
library(ggplot2)
library(multtest)
library(cowplot)
library(reshape2)
library(tidyr)
library(scales)
#Directories
DataPath <- '~/Documents/LegacyData/PlottingScripts/ProcessedData/'
figurePath <- '~/Documents/LegacyData/Figures/'
```
##StrictMask
```{bash}
DataPath='/Users/luke/Documents/LegacyData/PlottingScripts/ProcessedData/'

#from all positions create a pseudo BED file
cat $DataPath/rsID.txt | \
awk '{print $1,$2,$2+1,$3}' | \
tr ' ' '\t' \
> $DataPath/1kGP_pseudoBed.bed
#subtract the strict mask from the 1kGP
bedtools subtract \
-a $DataPath/1kGP_pseudoBed.bed \
-b $DataPath/20141020.strict_mask.whole_genome.bed \
> $DataPath/1kGP_StrictMask.bed
```
##Load Data
```{r}
##read Regression results
Regression <- fread(paste0(DataPath,'GenomeWide_LFA.txt'),
                    fill = TRUE,
                    col.names = c('rsID','devdiff','dev_logp','b'),
                    colClasses = c('character','numeric','numeric','numeric'))

##read Repeats
Repeats <- fread(paste0(DataPath,'Reg_strictMask_nestedRepeats.POS'),
                 fill = TRUE,
                 col.names = c('Chr','Pos'))
Repeats$Repeat='yes'

##read Indels
Indels <- fread(paste0(DataPath,'Reg_indels.POS'),
                header = TRUE)
Indels$Indel = 'yes'

#read in rsIDs (used for GWAS catalogue search)
rsID<-fread(paste0(DataPath,'rsID.txt'),
            col.names = c('Chr','Pos','rsID'))

strictMask <- fread(paste0(DataPath,
                           '1kGP_StrictMask.bed'),
            col.names = c('Chr','Pos','Pos.1','rsID'))
```


```{r}
##Combine the data
Regression <- left_join(Regression, 
                        rsID, 
                        by = 'rsID')

Regression <- left_join(Regression, 
                        Repeats, 
                        by = c('Chr','Pos'))

Regression <- left_join(Regression, 
                        Indels, 
                        by = c('Chr','Pos'))

#remove missing data
Regression <- Regression[which(!is.na(Regression$devdiff)),]
##save memory
rm(rsID)
rm(Repeats)
rm(Indels)
```

```{r}
Repeat_Indels <- unique(Regression[which(
                        Regression$Indel=='yes' &
								        Regression$Repeat=='yes'),])

NORepeat_Indels <- unique(Regression[which(
                        Regression$Indel=='yes' &
								        is.na(Regression$Repeat)),])
								
Repeat_SNPs <- unique(Regression[which(
                        is.na(Regression$Indel) &
								        Regression$Repeat=='yes'),])
								
NORepeat_SNPs <- unique(Regression[which(
                        is.na(Regression$Indel) &
								        is.na(Regression$Repeat)),])
NORepeat_SNPs <- NORepeat_SNPs[which(!is.na(NORepeat_SNPs$Chr)),]
```

```{r}
## two-stage Benjamini & Hochberg (2006) step-up FDR-controlling
TSBH <-function(df){	
procedures <- c( "TSBH") 

df$p <- pchisq(df$devdiff,
               1,
               lower.tail = FALSE)

adjusted <- mt.rawp2adjp(df$p, 
                         procedures, 
                         alpha = 0.01)

adj <- as.data.frame(adjusted$adj[
              order(adjusted$index),])

df$Adjusted_Log10p <- -log10(adj$TSBH_0.01)
df$Adjusted_p <- adj$TSBH_0.01

return(df)
}

Repeat_Indels <-TSBH(Repeat_Indels)
NORepeat_Indels <-TSBH(NORepeat_Indels)
Repeat_SNPs <-TSBH(Repeat_SNPs)
NORepeat_SNPs <-TSBH(NORepeat_SNPs)

Regression <- do.call("rbind", 
                      list(Repeat_Indels, 
                           NORepeat_Indels, 
                           Repeat_SNPs, 
                           NORepeat_SNPs))

#remove positions that don't pass strict mask
RegressionMasked <- anti_join(Regression, 
                        strictMask,
                        by = c('Chr','Pos','rsID'))

write.table(RegressionMasked[which(
            RegressionMasked$Adjusted_Log10p 
            > -log10(0.01)),] $ rsID, 
            file = paste0(DataPath, 
                          'Significant_rsID.txt'),
            quote = FALSE,
            row.names = FALSE,
            col.names = FALSE)

write.table(RegressionMasked[which(
            RegressionMasked$Adjusted_Log10p 
            > -log10(0.01)),c('Chr','Pos')], 
            file = paste0(DataPath, 
                          'Significant_Positions.txt'),
            quote = FALSE,
            row.names = FALSE,
            col.names = FALSE,
            sep = '\t')

write.table(RegressionMasked[which(
            RegressionMasked$Adjusted_Log10p 
            > -log10(0.01)),c('Chr','Pos','rsID','b','Adjusted_Log10p')], 
            file = paste0(DataPath, 
                          'Significant_Positions_and_Pvals.txt'),
            quote = FALSE,
            row.names = FALSE,
            sep = '\t')

nsigRepIndel <- nrow(unique(RegressionMasked[which(
                    RegressionMasked$Indel=='yes' &
                    RegressionMasked$Repeat=='yes' &
                    RegressionMasked$Adjusted_Log10p 
                    > -log10(0.01)),]))

nsigRepSNP <- nrow(unique(RegressionMasked[which(
                    is.na(RegressionMasked$Indel) &
                    RegressionMasked$Repeat=='yes' &
                    RegressionMasked$Adjusted_Log10p 
                    > -log10(0.01)),]))

nsigNonRepIndel <- nrow(unique(RegressionMasked[which(
                    RegressionMasked$Indel=='yes' &
                    is.na(RegressionMasked$Repeat) &
                    RegressionMasked$Adjusted_Log10p 
                    > -log10(0.01)),]))

nsigNonRepSNP <- nrow(unique(RegressionMasked[which(
                    is.na(RegressionMasked$Indel) &
                    is.na(RegressionMasked$Repeat) &
                    RegressionMasked$Adjusted_Log10p 
                    > -log10(0.01)),]))

print(paste0('Repeat Indels : ',nsigRepIndel,
             ' Repeat SNPs : ',nsigRepSNP,
             ' NonRepeat Indels : ',nsigNonRepIndel,
             ' NonRepeat SNPs : ',nsigNonRepSNP))
```


```{r}
makePlot <- function(df,Name){
sig.20<-df[which((df$Adjusted_Log10p >= 20)),]
sig.20$Adjusted_Log10p=20
sig.19<-df[which((df$Adjusted_Log10p < 20)),]
nSig <- nrow(df[which(df$Adjusted_Log10p > -log10(0.01)),])
             
ggplot(sig.19, 
       aes(x=Pos, 
           y = Adjusted_Log10p, 
           color=as.factor(Chr)))+
  geom_point(alpha = 0.3, 
             size = 1) +
  geom_point(data = sig.20, 
             aes(x = Pos, 
                 y = Adjusted_Log10p),
             shape = 1) +
  facet_grid(~Chr, 
             scales = 'free_x', 
             space = 'free_x', 
             switch = 'x') +
  scale_color_manual(values = rep(
                        c("grey", "black"), 
                        22)) +
  scale_y_continuous(expand = c(0,0)) +
  expand_limits(y=c(0,21))  +
  geom_hline(yintercept = -log10(0.01), 
             color = 'blue') +
  labs(y = '-log10(p)', 
       x = 'Chromosome', 
       title = Name, 
       subtitle = paste('Number of loci tested =',
                        nrow(df),
                        '\n',
                        'Number of significant loci =',
                        nSig)) +
  theme_classic() +
  theme(plot.title = element_text(hjust = 0.5), 
        plot.subtitle = element_text(hjust = 0.5), 
        axis.text.x = element_blank(), 
        axis.ticks.x = element_blank(),
        axis.line.x = element_blank(), 
        strip.background = element_blank(),
        strip.text.x = element_text(size = 6)) +
  guides(color = F)

dir=paste0(figurePath,
              'ManhattanPlot_')
ggsave(paste(dir,Name,'_strict.jpg',sep=''), height=5, width=10)
ggsave(paste(dir,Name,'_strict.tiff',sep=''), height=5, width=10)

}

makeQQ <- function(pvalues, name){
set.seed(1234)
pvalue <- runif(length(pvalues), min=0, max=1)
inflation <- function(ps) {
  chisq <- qchisq(1 - pvalue, 1)
  lambda <- median(chisq) / qchisq(0.5, 1)
  lambda
}
inflation(pvalue)

gg_qqplot(pvalues, name) +
  theme_bw(base_size = 24) +
  annotate(
    geom = "text",
    x = -Inf,
    y = Inf,
    hjust = -0.15,
    vjust = 1 + 0.15 * 3,
    label = sprintf("λ = %.2f", inflation(ps)),
    size = 8
  ) +
  theme(
    axis.ticks = element_line(size = 0.5),
    panel.grid = element_blank(),
    plot.title = element_text(hjust = 0.5)
  )

dir=paste0(figurePath,
              'QQPlot_')
ggsave(paste(dir,name,'_strict.jpg',sep=''), height=5, width=10)
ggsave(paste(dir,name,'_strict.tiff',sep=''), height=5, width=10)

 }
 ##modified from : https://slowkow.com/notes/ggplot2-qqplot/
 gg_qqplot <- function(ps,name) {
  n  <- length(ps)
  df <- data.frame(
    observed = -log10(sort(ps)),
    expected = -log10(ppoints(n)))
  log10Pe <- expression(paste("Expected -log"[10], plain(P)))
  log10Po <- expression(paste("Observed -log"[10], plain(P)))
  ggplot(df) +
    geom_point(aes(expected, observed), shape = 1, size = 3) +
    geom_abline(intercept = 0, slope = 1, alpha = 0.5) +
    xlab(log10Pe) +
    ylab(log10Po)
}

```
##Make the Manhattan Plots
```{r}
makePlot(Repeat_Indels, 
         'Repeat Indels')

makePlot(NORepeat_Indels, 
         'Non Repeat Indels')

makePlot(Repeat_SNPs, 
         'Repeat SNPs')

makePlot(NORepeat_SNPs, 
         'Non Repeat SNPs')
```
#Make the QQ Plots
```{r}
makeQQ(Repeat_Indels$p, 
       'Repeat Indels')

makeQQ(NORepeat_Indels$p, 
       'Non Repeat Indels')

makeQQ(Repeat_SNPs$p, 
       'Repeat SNPs')

makeQQ(NORepeat_SNPs $p, 
       'Non Repeat SNPs')
```

## GWAS publications
```{r}
catalog <- fread(paste0(DataPath,
                        'gwas_catalog_cut.txt'),
                 fill = TRUE,
                 sep = '\t')
catalog$rsID <- catalog$SNPS

catalog <- merge(catalog, Regression[,c('Chr','Pos', 'rsID','Adjusted_Log10p')], by='rsID')

sigTable <- catalog[which(catalog$Adjusted_Log10p > -log10(0.01)),
                    c('PUBMEDID','JOURNAL','rsID',
                      'PVALUE_MLOG','Adjusted_Log10p')]
sigTable <- sigTable[order(-sigTable$Adjusted_Log10p),]
sigTable$PVALUE_MLOG <- round(sigTable$PVALUE_MLOG,2)
sigTable$Adjusted_Log10p <- round(sigTable$Adjusted_Log10p,2)
write.table(sigTable, 
            file = paste0(DataPath,
                          'SignificantPublications_table.txt'),
            row.names = FALSE, 
            quote = FALSE, 
            sep = ' & ')

table <- catalog[which(catalog$log10P_0.01 > -log10(0.01)),
                 c('PUBMEDID','JOURNAL','SNPS',
                   'PVALUE_MLOG','Adjusted_Log10p')]
print(table[order(-Adjusted_Log10p),])
```

##Density of Significant SNPs per Chromosome
```{r}

sigPerChrom <- as.data.frame(table(RegressionMasked[which(RegressionMasked$Adjusted_Log10p > -log10(0.01)),]$Chr))
PerChrom <- as.data.frame(table(RegressionMasked$Chr))
sigPerChrom <- merge(sigPerChrom, PerChrom, by='Var1')
sigPerChrom$density <- sigPerChrom$Freq.x/sigPerChrom$Freq.y

ggplot(sigPerChrom, 
       aes(x=Var1,y=density))+
  geom_bar(stat="identity")+
  labs(x='Chromsome',
       y='proportion of significant variants')

ggsave(paste0(figurePath,
                'CountPerChrom.jpg'), 
       height=5, 
       width=10)

ChromLength <- fread(paste0(DataPath,
                            'GRCh37_ChromLength.txt'),
                     sep='\t',
                     col.names = c('Chr','Length'))
plts <- list()
bins <- list()
for(chrom in seq(1,22,1)){
  print(chrom)
  df <- RegressionMasked[which((RegressionMasked$Chr == chrom) &
                           (RegressionMasked$Adjusted_Log10p 
                            > -log10(0.01))),
                   c('Chr','Pos')]
  mask <- Regression[which((Regression$Chr == chrom) &
                           (Regression$Adjusted_Log10p 
                            > -log10(0.01)) &
                             !(Regression$rsID %in% RegressionMasked$rsID)),
                   c('Chr','Pos')]
  
  l = ChromLength[which(ChromLength$Chr == chrom),
                  c('Length')][[1]]
  b = seq(1,l,by=10000)
  if(chrom == 1){
    plts[[chrom]]<-ggplot(mask, aes(x=Pos)) +
    geom_freqpoly(breaks = b, color = 'grey60') +
    geom_freqpoly(data = df, breaks = b) +
    ylim(0,50)+
    theme(axis.text.x = element_blank(),
          axis.ticks.x = element_blank(),
          axis.line = element_blank(),
          plot.margin = margin(0,0,0,0),
          axis.text=element_text(size=6),
          axis.title=element_text(size=8)) + 
    labs(x=chrom,y='Significant variants per 10kb window')
  }
  else{
  plts[[chrom]]<-ggplot(mask, aes(x=Pos)) +
    geom_freqpoly(breaks = b, color = 'grey60') +
    geom_freqpoly(data = df, breaks = b) +
    ylim(0,50)+
    theme(axis.text = element_blank(),
          axis.ticks = element_blank(),
          axis.title.y = element_blank(),
          axis.line = element_blank(),
          plot.margin = margin(0,0,0,0),
          axis.title=element_text(size=8)) + 
    xlab(chrom)
  }
  #get freq count per bin
  c <- cut(df$Pos, breaks = b)
  bins[[chrom]] <- as.data.frame(table(c))
}

 TotalBins <- bins %>% bind_rows(.id = "source")
 ppHist <- ggplot(TotalBins, aes(x = Freq)) +
            geom_histogram(binwidth=1) +
            scale_y_log10() +
            theme_classic() +
            labs(x='Number of Q-associated variants in a 10kb window',
                 y='log10 Frequency')
            
relWidths <- unlist(ChromLength$Length)/100000000
relWidths[[1]] <- relWidths[[1]]*1.5
p <- plot_grid(plts[[1]],plts[[2]],plts[[3]],plts[[4]],plts[[5]],plts[[6]],plts[[7]],plts[[8]],plts[[9]],plts[[10]],plts[[11]],plts[[12]],plts[[13]],plts[[14]],plts[[15]],plts[[16]],plts[[17]],plts[[18]],plts[[19]],plts[[20]],plts[[21]],plts[[22]],nrow=1,rel_widths = relWidths)

p2 <- plot_grid(p, ppHist,nrow=2)

ggsave(paste0(figurePath,
                'VariantDensity.jpg'), 
       height=5, 
       width=10)

#get 10kb regions with more than 10 sig snps
over10  <- bins %>%
   bind_rows(.id = "source") %>%
   filter(Freq >= 5) %>%
   separate(c, c("start_bp", "end_bp"), ",")

over10$start_bp <- as.numeric(as.character(substring(over10$start_bp,2)))
over10$end_bp <- as.numeric(as.character(substring(over10$end_bp,1,
                                                    nchar(over10$end_bp)-1)))
names(over10)[c(1,4)]<-c('Chr','Freq_Sig_Var')
write.table(over10,
            file = paste0(DataPath,
                          'RegionsOver10.bed'),
            row.names = FALSE, 
            quote = FALSE,
            sep = '\t')

 makeSubPlot <- function(df,Name){
p<-ggplot(df, 
       aes(x=Pos, 
           y = Adjusted_Log10p))+
  geom_point(size = 1) +
  expand_limits(y=c(0,max(df$Adjusted_Log10p)+1))  +
  geom_hline(yintercept = -log10(0.01), 
             color = 'blue') +
  labs(y = '-log10(p)', 
       title = Name) +
  theme_classic() +
  theme(plot.title = element_text(hjust = 0.5), 
        axis.text.x = element_blank(), 
        axis.ticks.x = element_blank())

return(p)
}

top10<-head(over10[order(-over10$Freq),],10)
 
 sub <- list()
 pp <- list()
 for(i in seq(1,nrow(top10))){
   chr <- top10[i,]$Chr
   start <- top10[i,]$start_bp
   end <- top10[i,]$end_bp
   title <- paste0('Chr',chr, ':',start,'-',end)
   sub[[i]] <- RegressionMasked[which((RegressionMasked$Chr == chr) &
                              (RegressionMasked$Pos > start) &
                              (RegressionMasked$Pos < end)),
                              c('Chr','Pos','rsID',
                                'Adjusted_Log10p',
                                'Repeat','Indel')]
   pp[[i]] <- makeSubPlot(sub[[i]],title)
 }

 pp[[1]]
  ggsave(paste0(figurePath,
                'RegionOver10.jpg'), 
       height=5, 
       width=8)
  
 ggsave(paste0(figurePath,
                'top10.jpg'), 
       height=12, 
       width=8)
 
 
 library(Homo.sapiens)
 library(annotate)
 mycoords.gr = over10 %>%
   dplyr::select(chrom=Chr, start=start_bp, end=end_bp) %>%
   mutate(chrom=paste0('chr', chrom)) %>%
   makeGRangesFromDataFrame
 
 GeneRegions <- subsetByOverlaps(genes(TxDb.Hsapiens.UCSC.hg19.knownGene), mycoords.gr)
 
out <- data.frame(seqnames=seqnames(GeneRegions),
  starts=start(GeneRegions)-1,
  ends=end(GeneRegions),
  names=c(rep(".", length(GeneRegions))),
  scores=c(rep(".", length(GeneRegions))),
  strands=strand(GeneRegions),
  gene_id=GeneRegions$gene_id)

out$gene_symbol <- getSYMBOL(as.character(unlist(out$gene_id)), 
                        data='org.Hs.eg')

write.table(out, file = paste0(DataPath,
                          'RegionsOver10_geneID.bed'), 
                          quote=F, 
                          sep="\t", 
                          row.names=F)
```
