#  Download data and convert to Plink format
```{bash}
GenoName='phase3_shapeit2_mvncall_integrated_v5a.20130502'
GenoPath='/Users/luke/genomes/genomes/hg19/phase3'
PlinkPath='/Users/luke/genomes/genomes/hg19/plink'
PlinkDir='/Users/luke/bin/plink_mac'
cd $GenoPath

for f in `seq 1 22`
  do echo $f
  if [ ! -f $GenoPath/ALL.chr${f}.${GenoName}.genotypes.vcf.gz ]; then
    echo "Downloading VCF CHR ${f}"
    wget ftp://ftp-trace.ncbi.nih.gov/1000genomes/ftp/release/20130502/ALL.chr${f}.${GenoName}.genotypes.vcf.gz
  fi

  if [ ! -f $PlinkPath/chr${f}.bed ]; then
    echo "Converting VCF to Plink : Chr ${f}"
    $PlinkDir/plink \
    --vcf $GenoPath/ALL.chr${f}.${GenoName}.genotypes.vcf.gz \
    --make-bed \
    --out $PlinkPath/chr${f}.bed
  fi
```