---
title: "HGP analysis: fit model"
author: 
- name: Rick Farouni, 
  affiliation:
  - &cruk Department of Human Genetics, McGill University,  Montreal, Canada
date: '`r format(Sys.Date(), "%Y-%B-%d")`'
output:
  html_notebook:
    df_print: paged
    code_folding: show
    toc: yes
    toc_float: 
      collapsed: false
      smooth_scroll: false
---
  
  
  # Preparations
  
```{r setup}
args <- commandArgs(trailingOnly=T)

#path <- args[1]
path <- '/Users/luke/Documents/QualityPaper/rick/final/'
setwd(path)

#plinkPath <- args[2] 
plinkPath <- '/Users/luke/Documents/QualityPaper/ProcessedData' #'/Users/luke/genomes/genomes/hg19/plink/'

#chr <- args[3]
chr <- 'PublishedVariants'

knitr::opts_knit$set(root.dir = rprojroot::find_rstudio_root_file(), fig.width=16)
options(scipen=999)
```

```{r}
library(tidyverse)
library(lfa)
library(furrr)
library(glm2)
library(tictoc)
library(data.table)
library(cowplot)
options(future.globals.maxSize= 25000000000)
plan(multicore)
```
load plink format genotype data
```{r}
data_genotype <- read.bed(paste0(plinkPath,'chr',chr))

message("plink data loaded")
```
get rsIDs and add them as rownames for genotype data
```{r}
data_bim <- fread(paste0(plinkPath,'chr',chr,'.bim'), 
                  col.names = c('Chr','rsID','X',
                                'Pos','Ref','Alt'))

data_bim <- 
  data_bim %>%  
  rowid_to_column() %>% 
  mutate(rsID= if_else(!duplicated(rsID),
                       rsID,
                       paste(rsID,
                             rowid,
                             sep="_" ))) %>% 
  select(-rowid)

rownames(data_genotype) <- data_bim$rsID

#data_genotype <- head(data_genotype, 50000)
```
run lfa if needed
```{r lfa}
tic()
#LF <- lfa(data_genotype, 5)
toc()
message("lfa run complete")
```
save lfa if needed
```{r save}
tic()
#saveRDS(LF, "./data/preprocessed/lf5.rds")
toc()
```
load logistic factor analysis results
```{r read lf}
LF <- readRDS( "./data/preprocessed/lf5.rds")
```

define functions
```{r}
#this runs the logistic regression
lreg <- function(y, X){
  X <- rbind(X, X)
  y1 <- as.numeric((y==1) | (y==2))
  y2 <- as.numeric(y==2)
  y <- c(y1,y2)
  b <- glm2(cbind(y, 2-y) ~ -1 + X, family="binomial")$coef
  
  return(b)
  
}
#modified from gcat
fit_glm <- function(..., LF, trait){
  snp <- c(...)
  ind <- !is.na(snp) #logical index vector
  snp_no_na <- snp[ind]
  LF_no_na <- LF[ind,]
  p0 <- rep(NA, length(snp))
  p1 <- rep(NA, length(snp))
  trait_no_na <- trait[ind]
  
  b0 <- lreg(snp_no_na, LF_no_na) #coefficients from logreg 
  b1 <- lreg(snp_no_na, cbind(LF_no_na, trait_no_na))
  est0 <- .Call("mv", LF_no_na, b0)
  est1 <- .Call("mv", cbind(LF_no_na, trait_no_na), b1)
  p0[ind] <- exp(est0)/(1+exp(est0))
  p1[ind] <- exp(est1)/(1+exp(est1))
  
  dataset <-
    tibble(snp=snp,
           p0 = p0,
           p1 = p1)
  
  stat <-
    dataset %>%
    summarize(
      dev0 = -2 * sum(snp * log(p0) +
                        (2 - snp) * log(1 - p0),
                      na.rm = TRUE),
      dev1 = -2 * sum(snp * log(p1) +
                        (2 - snp) * log(1 - p1),
                      na.rm = TRUE),
      devdiff = dev0 - dev1,
      dev_logp = -log10(pchisq(devdiff,
                               1,
                               lower.tail = FALSE))
    ) %>% 
    mutate(num_nas = sum(!ind),
	   num_zeros= sum(snp_no_na ==0),
           b=b1[length(b1)])
  
  return(list(dataset=dataset, stat=stat))
         
}
```

```{r}
gc()
```

```{r}
fam_file <-fread(paste0(plinkPath,'chr',chr,'.fam'), 
                  col.names=c('sample','a','b','c','d','e'))
                 
bas_filepath <- "./data/metadata/average_quality_dt.txt"
bas_dt <- read_tsv(bas_filepath)
bas_dt <- merge(fam_file, 
                bas_dt,
                by = 'sample') 
avg_qual_mean <- bas_dt$avg_qual_mean
```

```{r}
head(avg_qual_mean)
```
#LFA
```{r, warning=FALSE}
tic("fitting model")
labels <- rownames(data_genotype)
data_fit <- 
	data_genotype %>%
	as_tibble() %>%
	transmute(
	fit = future_pmap(., fit_glm,
	LF=LF, 
	trait=avg_qual_mean),
		rsID=labels)  %>%
	unnest(fit) %>%
	mutate(l= rep(c("data","stat"), n()/2))%>%
	spread(l, fit)%>%
	unnest(stat)
 
saveRDS(data_fit, 
	file.path("./data/preprocessed/subsets",
	paste0("fits_chrom_", chr, "_LFA.rds")))

write.table(data_fit[-2], 
              file.path("./data/preprocessed/subsets", 
                        paste0("fits_chrom_", chr, "_LFA.txt")),
              quote = FALSE,
              row.names = FALSE)
rm(data_fit)
```
#PCA
```{r, warning=FALSE}
#load PCA
library(data.table)
samples = fread('/Users/luke/Documents/PCAperPop/Name_Pop_Qual_PC1_PC2_PC3_PC4_PC5.txt')
PC <- as.matrix(samples[,c('PC1','PC2','PC3','PC4','PC5')])
PC <- cbind(PC, 1)

tic("fitting model")
labels <- rownames(data_genotype)
data_fit_PCA <- 
	data_genotype %>%
	as_tibble() %>%
	transmute(
	fit = future_pmap(., fit_glm,
	LF= PC, 
	trait=avg_qual_mean),
		rsID=labels)  %>%
	unnest(fit) %>%
	mutate(l= rep(c("data","stat"), n()/2))%>%
	spread(l, fit)%>%
	unnest(stat)

saveRDS(data_fit_PCA, 
	file.path("./data/preprocessed/subsets",
	paste0("fits_chrom_", chr, "_PCA.rds")))

write.table(data_fit_PCA[-2], 
              file.path("./data/preprocessed/subsets", 
                        paste0("fits_chrom_", chr, "_PCA.txt")),
              quote = FALSE,
              row.names = FALSE)
rm(data_fit_PCA)
```
#Population
```{r, warning=FALSE}

testModel <- model.matrix(data = bas_dt, object = ~Population)  

tic("fitting model")
labels <- rownames(data_genotype)
data_fit_Pop <- 
	data_genotype %>%
	as_tibble() %>%
	transmute(
	fit = future_pmap(., fit_glm,
	LF= testModel, 
	trait=avg_qual_mean),
		rsID=labels)  %>%
	unnest(fit) %>%
	mutate(l= rep(c("data","stat"), n()/2))%>%
	spread(l, fit)%>%
	unnest(stat)
   toc()

saveRDS(data_fit_Pop, 
	file.path("./data/preprocessed/subsets",
	paste0("fits_chrom_", chr, "_Pop.rds")))

write.table(data_fit_Pop[-2], 
              file.path("./data/preprocessed/subsets", 
                        paste0("fits_chrom_", chr, "_Pop.txt")),
              quote = FALSE,
              row.names = FALSE)

rm(data_fit_Pop)
```

```{r}
gc()
```

```{r}
sessionInfo()
```